{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "txt_summ_test1.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "authorship_tag": "ABX9TyPcYkMfErItQslL2m1lGdxA",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/youneslebboukh/text-summarization/blob/main/txt_summ_test1.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "o47yrDdeeuwq",
        "outputId": "1c50928e-9f92-477d-a114-50593a94fff7"
      },
      "source": [
        "!pip install kaggle"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: kaggle in /usr/local/lib/python3.6/dist-packages (1.5.9)\n",
            "Requirement already satisfied: urllib3 in /usr/local/lib/python3.6/dist-packages (from kaggle) (1.24.3)\n",
            "Requirement already satisfied: six>=1.10 in /usr/local/lib/python3.6/dist-packages (from kaggle) (1.15.0)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.6/dist-packages (from kaggle) (4.41.1)\n",
            "Requirement already satisfied: slugify in /usr/local/lib/python3.6/dist-packages (from kaggle) (0.0.1)\n",
            "Requirement already satisfied: python-dateutil in /usr/local/lib/python3.6/dist-packages (from kaggle) (2.8.1)\n",
            "Requirement already satisfied: python-slugify in /usr/local/lib/python3.6/dist-packages (from kaggle) (4.0.1)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.6/dist-packages (from kaggle) (2.23.0)\n",
            "Requirement already satisfied: certifi in /usr/local/lib/python3.6/dist-packages (from kaggle) (2020.11.8)\n",
            "Requirement already satisfied: text-unidecode>=1.3 in /usr/local/lib/python3.6/dist-packages (from python-slugify->kaggle) (1.3)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.6/dist-packages (from requests->kaggle) (3.0.4)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests->kaggle) (2.10)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "resources": {
            "http://localhost:8080/nbextensions/google.colab/files.js": {
              "data": "Ly8gQ29weXJpZ2h0IDIwMTcgR29vZ2xlIExMQwovLwovLyBMaWNlbnNlZCB1bmRlciB0aGUgQXBhY2hlIExpY2Vuc2UsIFZlcnNpb24gMi4wICh0aGUgIkxpY2Vuc2UiKTsKLy8geW91IG1heSBub3QgdXNlIHRoaXMgZmlsZSBleGNlcHQgaW4gY29tcGxpYW5jZSB3aXRoIHRoZSBMaWNlbnNlLgovLyBZb3UgbWF5IG9idGFpbiBhIGNvcHkgb2YgdGhlIExpY2Vuc2UgYXQKLy8KLy8gICAgICBodHRwOi8vd3d3LmFwYWNoZS5vcmcvbGljZW5zZXMvTElDRU5TRS0yLjAKLy8KLy8gVW5sZXNzIHJlcXVpcmVkIGJ5IGFwcGxpY2FibGUgbGF3IG9yIGFncmVlZCB0byBpbiB3cml0aW5nLCBzb2Z0d2FyZQovLyBkaXN0cmlidXRlZCB1bmRlciB0aGUgTGljZW5zZSBpcyBkaXN0cmlidXRlZCBvbiBhbiAiQVMgSVMiIEJBU0lTLAovLyBXSVRIT1VUIFdBUlJBTlRJRVMgT1IgQ09ORElUSU9OUyBPRiBBTlkgS0lORCwgZWl0aGVyIGV4cHJlc3Mgb3IgaW1wbGllZC4KLy8gU2VlIHRoZSBMaWNlbnNlIGZvciB0aGUgc3BlY2lmaWMgbGFuZ3VhZ2UgZ292ZXJuaW5nIHBlcm1pc3Npb25zIGFuZAovLyBsaW1pdGF0aW9ucyB1bmRlciB0aGUgTGljZW5zZS4KCi8qKgogKiBAZmlsZW92ZXJ2aWV3IEhlbHBlcnMgZm9yIGdvb2dsZS5jb2xhYiBQeXRob24gbW9kdWxlLgogKi8KKGZ1bmN0aW9uKHNjb3BlKSB7CmZ1bmN0aW9uIHNwYW4odGV4dCwgc3R5bGVBdHRyaWJ1dGVzID0ge30pIHsKICBjb25zdCBlbGVtZW50ID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3BhbicpOwogIGVsZW1lbnQudGV4dENvbnRlbnQgPSB0ZXh0OwogIGZvciAoY29uc3Qga2V5IG9mIE9iamVjdC5rZXlzKHN0eWxlQXR0cmlidXRlcykpIHsKICAgIGVsZW1lbnQuc3R5bGVba2V5XSA9IHN0eWxlQXR0cmlidXRlc1trZXldOwogIH0KICByZXR1cm4gZWxlbWVudDsKfQoKLy8gTWF4IG51bWJlciBvZiBieXRlcyB3aGljaCB3aWxsIGJlIHVwbG9hZGVkIGF0IGEgdGltZS4KY29uc3QgTUFYX1BBWUxPQURfU0laRSA9IDEwMCAqIDEwMjQ7CgpmdW5jdGlvbiBfdXBsb2FkRmlsZXMoaW5wdXRJZCwgb3V0cHV0SWQpIHsKICBjb25zdCBzdGVwcyA9IHVwbG9hZEZpbGVzU3RlcChpbnB1dElkLCBvdXRwdXRJZCk7CiAgY29uc3Qgb3V0cHV0RWxlbWVudCA9IGRvY3VtZW50LmdldEVsZW1lbnRCeUlkKG91dHB1dElkKTsKICAvLyBDYWNoZSBzdGVwcyBvbiB0aGUgb3V0cHV0RWxlbWVudCB0byBtYWtlIGl0IGF2YWlsYWJsZSBmb3IgdGhlIG5leHQgY2FsbAogIC8vIHRvIHVwbG9hZEZpbGVzQ29udGludWUgZnJvbSBQeXRob24uCiAgb3V0cHV0RWxlbWVudC5zdGVwcyA9IHN0ZXBzOwoKICByZXR1cm4gX3VwbG9hZEZpbGVzQ29udGludWUob3V0cHV0SWQpOwp9CgovLyBUaGlzIGlzIHJvdWdobHkgYW4gYXN5bmMgZ2VuZXJhdG9yIChub3Qgc3VwcG9ydGVkIGluIHRoZSBicm93c2VyIHlldCksCi8vIHdoZXJlIHRoZXJlIGFyZSBtdWx0aXBsZSBhc3luY2hyb25vdXMgc3RlcHMgYW5kIHRoZSBQeXRob24gc2lkZSBpcyBnb2luZwovLyB0byBwb2xsIGZvciBjb21wbGV0aW9uIG9mIGVhY2ggc3RlcC4KLy8gVGhpcyB1c2VzIGEgUHJvbWlzZSB0byBibG9jayB0aGUgcHl0aG9uIHNpZGUgb24gY29tcGxldGlvbiBvZiBlYWNoIHN0ZXAsCi8vIHRoZW4gcGFzc2VzIHRoZSByZXN1bHQgb2YgdGhlIHByZXZpb3VzIHN0ZXAgYXMgdGhlIGlucHV0IHRvIHRoZSBuZXh0IHN0ZXAuCmZ1bmN0aW9uIF91cGxvYWRGaWxlc0NvbnRpbnVlKG91dHB1dElkKSB7CiAgY29uc3Qgb3V0cHV0RWxlbWVudCA9IGRvY3VtZW50LmdldEVsZW1lbnRCeUlkKG91dHB1dElkKTsKICBjb25zdCBzdGVwcyA9IG91dHB1dEVsZW1lbnQuc3RlcHM7CgogIGNvbnN0IG5leHQgPSBzdGVwcy5uZXh0KG91dHB1dEVsZW1lbnQubGFzdFByb21pc2VWYWx1ZSk7CiAgcmV0dXJuIFByb21pc2UucmVzb2x2ZShuZXh0LnZhbHVlLnByb21pc2UpLnRoZW4oKHZhbHVlKSA9PiB7CiAgICAvLyBDYWNoZSB0aGUgbGFzdCBwcm9taXNlIHZhbHVlIHRvIG1ha2UgaXQgYXZhaWxhYmxlIHRvIHRoZSBuZXh0CiAgICAvLyBzdGVwIG9mIHRoZSBnZW5lcmF0b3IuCiAgICBvdXRwdXRFbGVtZW50Lmxhc3RQcm9taXNlVmFsdWUgPSB2YWx1ZTsKICAgIHJldHVybiBuZXh0LnZhbHVlLnJlc3BvbnNlOwogIH0pOwp9CgovKioKICogR2VuZXJhdG9yIGZ1bmN0aW9uIHdoaWNoIGlzIGNhbGxlZCBiZXR3ZWVuIGVhY2ggYXN5bmMgc3RlcCBvZiB0aGUgdXBsb2FkCiAqIHByb2Nlc3MuCiAqIEBwYXJhbSB7c3RyaW5nfSBpbnB1dElkIEVsZW1lbnQgSUQgb2YgdGhlIGlucHV0IGZpbGUgcGlja2VyIGVsZW1lbnQuCiAqIEBwYXJhbSB7c3RyaW5nfSBvdXRwdXRJZCBFbGVtZW50IElEIG9mIHRoZSBvdXRwdXQgZGlzcGxheS4KICogQHJldHVybiB7IUl0ZXJhYmxlPCFPYmplY3Q+fSBJdGVyYWJsZSBvZiBuZXh0IHN0ZXBzLgogKi8KZnVuY3Rpb24qIHVwbG9hZEZpbGVzU3RlcChpbnB1dElkLCBvdXRwdXRJZCkgewogIGNvbnN0IGlucHV0RWxlbWVudCA9IGRvY3VtZW50LmdldEVsZW1lbnRCeUlkKGlucHV0SWQpOwogIGlucHV0RWxlbWVudC5kaXNhYmxlZCA9IGZhbHNlOwoKICBjb25zdCBvdXRwdXRFbGVtZW50ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQob3V0cHV0SWQpOwogIG91dHB1dEVsZW1lbnQuaW5uZXJIVE1MID0gJyc7CgogIGNvbnN0IHBpY2tlZFByb21pc2UgPSBuZXcgUHJvbWlzZSgocmVzb2x2ZSkgPT4gewogICAgaW5wdXRFbGVtZW50LmFkZEV2ZW50TGlzdGVuZXIoJ2NoYW5nZScsIChlKSA9PiB7CiAgICAgIHJlc29sdmUoZS50YXJnZXQuZmlsZXMpOwogICAgfSk7CiAgfSk7CgogIGNvbnN0IGNhbmNlbCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2J1dHRvbicpOwogIGlucHV0RWxlbWVudC5wYXJlbnRFbGVtZW50LmFwcGVuZENoaWxkKGNhbmNlbCk7CiAgY2FuY2VsLnRleHRDb250ZW50ID0gJ0NhbmNlbCB1cGxvYWQnOwogIGNvbnN0IGNhbmNlbFByb21pc2UgPSBuZXcgUHJvbWlzZSgocmVzb2x2ZSkgPT4gewogICAgY2FuY2VsLm9uY2xpY2sgPSAoKSA9PiB7CiAgICAgIHJlc29sdmUobnVsbCk7CiAgICB9OwogIH0pOwoKICAvLyBXYWl0IGZvciB0aGUgdXNlciB0byBwaWNrIHRoZSBmaWxlcy4KICBjb25zdCBmaWxlcyA9IHlpZWxkIHsKICAgIHByb21pc2U6IFByb21pc2UucmFjZShbcGlja2VkUHJvbWlzZSwgY2FuY2VsUHJvbWlzZV0pLAogICAgcmVzcG9uc2U6IHsKICAgICAgYWN0aW9uOiAnc3RhcnRpbmcnLAogICAgfQogIH07CgogIGNhbmNlbC5yZW1vdmUoKTsKCiAgLy8gRGlzYWJsZSB0aGUgaW5wdXQgZWxlbWVudCBzaW5jZSBmdXJ0aGVyIHBpY2tzIGFyZSBub3QgYWxsb3dlZC4KICBpbnB1dEVsZW1lbnQuZGlzYWJsZWQgPSB0cnVlOwoKICBpZiAoIWZpbGVzKSB7CiAgICByZXR1cm4gewogICAgICByZXNwb25zZTogewogICAgICAgIGFjdGlvbjogJ2NvbXBsZXRlJywKICAgICAgfQogICAgfTsKICB9CgogIGZvciAoY29uc3QgZmlsZSBvZiBmaWxlcykgewogICAgY29uc3QgbGkgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaScpOwogICAgbGkuYXBwZW5kKHNwYW4oZmlsZS5uYW1lLCB7Zm9udFdlaWdodDogJ2JvbGQnfSkpOwogICAgbGkuYXBwZW5kKHNwYW4oCiAgICAgICAgYCgke2ZpbGUudHlwZSB8fCAnbi9hJ30pIC0gJHtmaWxlLnNpemV9IGJ5dGVzLCBgICsKICAgICAgICBgbGFzdCBtb2RpZmllZDogJHsKICAgICAgICAgICAgZmlsZS5sYXN0TW9kaWZpZWREYXRlID8gZmlsZS5sYXN0TW9kaWZpZWREYXRlLnRvTG9jYWxlRGF0ZVN0cmluZygpIDoKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgJ24vYSd9IC0gYCkpOwogICAgY29uc3QgcGVyY2VudCA9IHNwYW4oJzAlIGRvbmUnKTsKICAgIGxpLmFwcGVuZENoaWxkKHBlcmNlbnQpOwoKICAgIG91dHB1dEVsZW1lbnQuYXBwZW5kQ2hpbGQobGkpOwoKICAgIGNvbnN0IGZpbGVEYXRhUHJvbWlzZSA9IG5ldyBQcm9taXNlKChyZXNvbHZlKSA9PiB7CiAgICAgIGNvbnN0IHJlYWRlciA9IG5ldyBGaWxlUmVhZGVyKCk7CiAgICAgIHJlYWRlci5vbmxvYWQgPSAoZSkgPT4gewogICAgICAgIHJlc29sdmUoZS50YXJnZXQucmVzdWx0KTsKICAgICAgfTsKICAgICAgcmVhZGVyLnJlYWRBc0FycmF5QnVmZmVyKGZpbGUpOwogICAgfSk7CiAgICAvLyBXYWl0IGZvciB0aGUgZGF0YSB0byBiZSByZWFkeS4KICAgIGxldCBmaWxlRGF0YSA9IHlpZWxkIHsKICAgICAgcHJvbWlzZTogZmlsZURhdGFQcm9taXNlLAogICAgICByZXNwb25zZTogewogICAgICAgIGFjdGlvbjogJ2NvbnRpbnVlJywKICAgICAgfQogICAgfTsKCiAgICAvLyBVc2UgYSBjaHVua2VkIHNlbmRpbmcgdG8gYXZvaWQgbWVzc2FnZSBzaXplIGxpbWl0cy4gU2VlIGIvNjIxMTU2NjAuCiAgICBsZXQgcG9zaXRpb24gPSAwOwogICAgd2hpbGUgKHBvc2l0aW9uIDwgZmlsZURhdGEuYnl0ZUxlbmd0aCkgewogICAgICBjb25zdCBsZW5ndGggPSBNYXRoLm1pbihmaWxlRGF0YS5ieXRlTGVuZ3RoIC0gcG9zaXRpb24sIE1BWF9QQVlMT0FEX1NJWkUpOwogICAgICBjb25zdCBjaHVuayA9IG5ldyBVaW50OEFycmF5KGZpbGVEYXRhLCBwb3NpdGlvbiwgbGVuZ3RoKTsKICAgICAgcG9zaXRpb24gKz0gbGVuZ3RoOwoKICAgICAgY29uc3QgYmFzZTY0ID0gYnRvYShTdHJpbmcuZnJvbUNoYXJDb2RlLmFwcGx5KG51bGwsIGNodW5rKSk7CiAgICAgIHlpZWxkIHsKICAgICAgICByZXNwb25zZTogewogICAgICAgICAgYWN0aW9uOiAnYXBwZW5kJywKICAgICAgICAgIGZpbGU6IGZpbGUubmFtZSwKICAgICAgICAgIGRhdGE6IGJhc2U2NCwKICAgICAgICB9LAogICAgICB9OwogICAgICBwZXJjZW50LnRleHRDb250ZW50ID0KICAgICAgICAgIGAke01hdGgucm91bmQoKHBvc2l0aW9uIC8gZmlsZURhdGEuYnl0ZUxlbmd0aCkgKiAxMDApfSUgZG9uZWA7CiAgICB9CiAgfQoKICAvLyBBbGwgZG9uZS4KICB5aWVsZCB7CiAgICByZXNwb25zZTogewogICAgICBhY3Rpb246ICdjb21wbGV0ZScsCiAgICB9CiAgfTsKfQoKc2NvcGUuZ29vZ2xlID0gc2NvcGUuZ29vZ2xlIHx8IHt9OwpzY29wZS5nb29nbGUuY29sYWIgPSBzY29wZS5nb29nbGUuY29sYWIgfHwge307CnNjb3BlLmdvb2dsZS5jb2xhYi5fZmlsZXMgPSB7CiAgX3VwbG9hZEZpbGVzLAogIF91cGxvYWRGaWxlc0NvbnRpbnVlLAp9Owp9KShzZWxmKTsK",
              "ok": true,
              "headers": [
                [
                  "content-type",
                  "application/javascript"
                ]
              ],
              "status": 200,
              "status_text": ""
            }
          },
          "base_uri": "https://localhost:8080/",
          "height": 89
        },
        "id": "b4C6_7Z3fIN3",
        "outputId": "af5063d0-ef57-4527-8566-b16e5d2c8802"
      },
      "source": [
        "from google.colab import files\n",
        "files.upload()"
      ],
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "\n",
              "     <input type=\"file\" id=\"files-e89321e5-2479-4ed3-b862-2020864dddd0\" name=\"files[]\" multiple disabled\n",
              "        style=\"border:none\" />\n",
              "     <output id=\"result-e89321e5-2479-4ed3-b862-2020864dddd0\">\n",
              "      Upload widget is only available when the cell has been executed in the\n",
              "      current browser session. Please rerun this cell to enable.\n",
              "      </output>\n",
              "      <script src=\"/nbextensions/google.colab/files.js\"></script> "
            ],
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "stream",
          "text": [
            "Saving kaggle.json to kaggle.json\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'kaggle.json': b'{\"username\":\"youneslbk\",\"key\":\"ff30355568a25aef64e5c064a14e36fc\"}'}"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 2
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "RId33NszftMP"
      },
      "source": [
        " !mkdir -p ~/.kaggle\n",
        " !cp kaggle.json ~/.kaggle/"
      ],
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_mRQTOEkgZJ9"
      },
      "source": [
        "  !chmod 600 ~/.kaggle/kaggle.json"
      ],
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QcvBdXSAg0Bv",
        "outputId": "7f13abd8-aaf3-4831-c6eb-ca45c41bbede"
      },
      "source": [
        "!kaggle datasets download -d allen-institute-for-ai/CORD-19-research-challenge"
      ],
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Downloading CORD-19-research-challenge.zip to /content\n",
            "100% 6.18G/6.19G [01:30<00:00, 39.8MB/s]\n",
            "100% 6.19G/6.19G [01:30<00:00, 73.6MB/s]\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ivtbsfgvhczW",
        "outputId": "5da625d5-6054-4b5f-cb22-e39f2a1dcf73"
      },
      "source": [
        "!pip install zip_files"
      ],
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Collecting zip_files\n",
            "  Downloading https://files.pythonhosted.org/packages/f3/99/fc7f65a052d30e62b946924ed6334f8f1ebf7c826646b9e17ab1dff2901c/zip_files-0.3.0-py2.py3-none-any.whl\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.6/dist-packages (from zip_files) (7.1.2)\n",
            "Installing collected packages: zip-files\n",
            "Successfully installed zip-files-0.3.0\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "s2QufpcThhCC",
        "outputId": "f4c2efbe-efb9-452e-e251-c2d03960d1be"
      },
      "source": [
        "from zipfile import ZipFile\n",
        "file_name=\"CORD-19-research-challenge.zip\"\n",
        "with ZipFile(file_name,'r') as zip:\n",
        "  zip.extractall()\n",
        "  print('Done')"
      ],
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Done\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q6wBDJin_0FI",
        "outputId": "2eabed35-a8d4-401b-9d0b-047a85e608dd"
      },
      "source": [
        "import numpy as np\n",
        "import pandas as pd\n",
        "import nltk\n",
        "nltk.download('punkt') # one time execution\n",
        "import re"
      ],
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[nltk_data] Downloading package punkt to /root/nltk_data...\n",
            "[nltk_data]   Package punkt is already up-to-date!\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PFg83UK0BC77",
        "outputId": "7ecdd6fc-28e0-4b91-98a0-7033826d1cee"
      },
      "source": [
        "metadata = pd.read_csv(\"/content/metadata.csv\")"
      ],
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (1,4,5,6,13,14,15,16) have mixed types.Specify dtype option on import or set low_memory=False.\n",
            "  interactivity=interactivity, compiler=compiler, result=result)\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 564
        },
        "id": "uV4DtP5-BTaA",
        "outputId": "e33b5f24-f1b0-4e49-ff65-a2362d9c98f9"
      },
      "source": [
        "metadata.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>cord_uid</th>\n",
              "      <th>sha</th>\n",
              "      <th>source_x</th>\n",
              "      <th>title</th>\n",
              "      <th>doi</th>\n",
              "      <th>pmcid</th>\n",
              "      <th>pubmed_id</th>\n",
              "      <th>license</th>\n",
              "      <th>abstract</th>\n",
              "      <th>publish_time</th>\n",
              "      <th>authors</th>\n",
              "      <th>journal</th>\n",
              "      <th>mag_id</th>\n",
              "      <th>who_covidence_id</th>\n",
              "      <th>arxiv_id</th>\n",
              "      <th>pdf_json_files</th>\n",
              "      <th>pmc_json_files</th>\n",
              "      <th>url</th>\n",
              "      <th>s2_id</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ug7v899j</td>\n",
              "      <td>d1aafb70c066a2068b02786f8929fd9c900897fb</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
              "      <td>10.1186/1471-2334-1-6</td>\n",
              "      <td>PMC35282</td>\n",
              "      <td>11472636</td>\n",
              "      <td>no-cc</td>\n",
              "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
              "      <td>2001-07-04</td>\n",
              "      <td>Madani, Tariq A; Al-Ghamdi, Aisha A</td>\n",
              "      <td>BMC Infect Dis</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/d1aafb70c066a2068b027...</td>\n",
              "      <td>document_parses/pmc_json/PMC35282.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>02tnwd4m</td>\n",
              "      <td>6b0567729c2143a66d737eb0a2f63f2dce2e5a7d</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
              "      <td>10.1186/rr14</td>\n",
              "      <td>PMC59543</td>\n",
              "      <td>11667967</td>\n",
              "      <td>no-cc</td>\n",
              "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
              "      <td>2000-08-15</td>\n",
              "      <td>Vliet, Albert van der; Eiserich, Jason P; Cros...</td>\n",
              "      <td>Respir Res</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/6b0567729c2143a66d737...</td>\n",
              "      <td>document_parses/pmc_json/PMC59543.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ejv2xln0</td>\n",
              "      <td>06ced00a5fc04215949aa72528f2eeaae1d58927</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
              "      <td>10.1186/rr19</td>\n",
              "      <td>PMC59549</td>\n",
              "      <td>11667972</td>\n",
              "      <td>no-cc</td>\n",
              "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
              "      <td>2000-08-25</td>\n",
              "      <td>Crouch, Erika C</td>\n",
              "      <td>Respir Res</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/06ced00a5fc04215949aa...</td>\n",
              "      <td>document_parses/pmc_json/PMC59549.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>2b73a28n</td>\n",
              "      <td>348055649b6b8cf2b9a376498df9bf41f7123605</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Role of endothelin-1 in lung disease</td>\n",
              "      <td>10.1186/rr44</td>\n",
              "      <td>PMC59574</td>\n",
              "      <td>11686871</td>\n",
              "      <td>no-cc</td>\n",
              "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
              "      <td>2001-02-22</td>\n",
              "      <td>Fagan, Karen A; McMurtry, Ivan F; Rodman, David M</td>\n",
              "      <td>Respir Res</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/348055649b6b8cf2b9a37...</td>\n",
              "      <td>document_parses/pmc_json/PMC59574.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>9785vg6d</td>\n",
              "      <td>5f48792a5fa08bed9f56016f4981ae2ca6031b32</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Gene expression in epithelial cells in respons...</td>\n",
              "      <td>10.1186/rr61</td>\n",
              "      <td>PMC59580</td>\n",
              "      <td>11686888</td>\n",
              "      <td>no-cc</td>\n",
              "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
              "      <td>2001-05-11</td>\n",
              "      <td>Domachowske, Joseph B; Bonville, Cynthia A; Ro...</td>\n",
              "      <td>Respir Res</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/5f48792a5fa08bed9f560...</td>\n",
              "      <td>document_parses/pmc_json/PMC59580.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   cord_uid  ... s2_id\n",
              "0  ug7v899j  ...   NaN\n",
              "1  02tnwd4m  ...   NaN\n",
              "2  ejv2xln0  ...   NaN\n",
              "3  2b73a28n  ...   NaN\n",
              "4  9785vg6d  ...   NaN\n",
              "\n",
              "[5 rows x 19 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 18
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_Ek4HYZ-KsZ1"
      },
      "source": [
        "import json"
      ],
      "execution_count": 13,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "r8gRJlXTJubR"
      },
      "source": [
        "import glob"
      ],
      "execution_count": 14,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NUAv4K8eKGQl",
        "outputId": "6bfb362d-e9bb-4bde-f1f1-bbfedc57e973"
      },
      "source": [
        "p=\"/content/document_parses\"\r\n",
        "all_json = glob.glob(f'{p}/**/*.json', recursive=True)\r\n",
        "len(all_json)\r\n",
        "#print(all_json[0])"
      ],
      "execution_count": 63,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "240409"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 63
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "UTL0cVpCGPX0"
      },
      "source": [
        "full_body_text=[]\r\n",
        "for i in range(5) :\r\n",
        "  with open(all_json[i]) as file :\r\n",
        "    content = json.load(file)\r\n",
        "    abstract = []\r\n",
        "    body_text = []\r\n",
        "    # Body text\r\n",
        "    for entry in content['body_text']:\r\n",
        "      body_text.append(entry['text'])\r\n",
        "    full_body_text.append(body_text)\r\n",
        "    body_text = '\\n'.join(body_text)\r\n",
        "   \r\n",
        "      # Abstract\r\n",
        "    #for entry in content['abstract']:\r\n",
        "    # abstract.append(entry['text'])\r\n",
        "    #abstract = '\\n'.join(abstract)"
      ],
      "execution_count": 79,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5XHvqu4r9Bj8",
        "outputId": "b7938ba3-fa9f-4024-8119-5639fb0b6e5c"
      },
      "source": [
        "print(body_text)\r\n",
        "print(\"***************************************************************************************\")\r\n",
        "for i in full_body_text :\r\n",
        "  print(i)"
      ],
      "execution_count": 92,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "All viruses require host translational machinery in order to replicate. To make use of this machinery while avoiding detection, viruses have evolved a number of unconventional translational strategies (Gale et al, 2000), including internal ribosome entry sites (Jang et al, 1988), leaky scanning (Schwartz et al, 1990), ribosome shunting (Schmidt-Puchta et al, 1997), reinitiation (Hemmings-Mieszczak et al, 1997), and programmed ribosomal frameshifting (PRF) (Gesteland and Atkins, 1996). Many of these strategies are unique to viruses, and thus are potential antiviral targets (Harford, 1995).\n",
            "During PRF, the translational machinery slips backward or forward by one nucleotide while decoding a so-called ‘slippery sequence,' a heptanucleotide sequence of the form XXXYYYZ in the mRNA transcript (Figure 1). Outside of PRF, frameshifting is a rare event, occurring fewer than 5 × 10−5 times per codon (Kurland, 1992). PRF can increase the occurrence of frameshifting by several orders of magnitude, depending on the slippery sequence itself as well as on the presence or absence of adjacent secondary structures that favor translational pausing and increase frameshift frequency (Gesteland and Atkins, 1996). In many instances of PRF, slippage causes the translational machinery to bypass a stop codon just downstream of the slippery sequence, resulting in the expression of an elongated protein (Gesteland and Atkins, 1996). Therefore, normal expression from these transcripts results in two products, a shorter product based on the initial translation frame plus a longer product, whose ratio depends on the frequency of slippage and is thought to have evolved to a value optimized for virus production (Shehu-Xhilaga et al, 2001; Dulude et al, 2006).\n",
            "PRF is an evolutionarily conserved phenomenon observed in a range of viruses, from bacteriophage to notable human pathogens (Table I; Baranov et al, 2002). For example, in HIV, the Gag to Gag-Pol expression ratio is regulated by PRF and is necessary for the proper loading of reverse transcriptase into the new viral capsids; disturbing this ratio can inhibit viral replication (Shehu-Xhilaga et al, 2001; Dulude et al, 2006). The role of PRF in the SARS life cycle is less clear, but is necessary for expression of its RNA-dependent RNA polymerase (Plant et al, 2010). In Escherichia coli bacteriophage lambda, PRF regulates the expression of proteins gpG and gpGT (Levin et al, 1993). As lambda phage genes G and T lie between its major tail gene (V) and its tape measure gene (H), they are thought to have a role in tail formation (Xu, 2001). However, proteins gpG and gpGT do not make up any of the final tail structure (Levin et al, 1993), and thus may act as chaperones or scaffolds for gpH, assisting in the assembly of the tail shaft around the tape measure protein (Xu, 2001).\n",
            "Reading-frame maintenance is a common function of tRNA modifications (Urbonavicius et al, 2001). tRNAs contain upwards of 80 known modified nucleosides (Limbach et al, 1994), and many of these are strongly conserved across species (Sprinzl and Vassilenko, 2005). However, their loss is generally not lethal (Alexandrov et al, 2006; Phizicky and Hopper, 2010). In E. coli, four tRNA modifications require the addition of a thiol moiety and are made via two functionally distinct mechanisms (Nilsson et al, 2002). The first mechanism requires proteins that contain iron-sulfur (Fe-S) clusters and includes 2-thiocytidine formation at position 32 and methylthio formation of ms2i6A at position 37 (Jager et al, 2004; Pierrel et al, 2004). The second mechanism is independent of Fe-S clusters and includes 4-thiouridine at position 8 and 2-thiouridine at position 34 (Ikeuchi et al, 2006). For each of these modifications, sulfur is sequestered by E. coli's primary cysteine desulfurase, IscS. Sulfur is then used for one of these four tRNA modifications or for Fe-S cluster biosynthesis. In addition to their roles in tRNA modification, Fe-S clusters are critical cofactors involved in many cellular processes including respiration, central metabolism, environmental sensing, RNA modification, DNA repair, and DNA replication (Py and Barras, 2010).\n",
            "We previously identified two pathways downstream of IscS in a screen to identify E. coli genes with a significant effect on lambda phage replication (Maynard et al, 2010). Specifically, lambda phage replication was inhibited following deletion of several members of the 2-thiouridine synthesis (TUS) pathway leading to 2-thiouridine modification of tRNALys/Glu/Gln, a pathway independent of Fe-S cluster biosynthesis. Conversely, we found that knocking out members of the Fe-S cluster biosynthesis (ISC) pathway enhanced lambda replication.\n",
            "The current investigation was therefore motivated by this overarching question: how do the host's TUS and ISC pathways control viral replication? We were particularly intrigued by the possibility that these pathways interact. Furthermore, the substantial evolutionary conservation of both host genes and viral PRF mechanism strongly suggested that our observations would have relevance beyond E. coli and lambda phage. Our current observations shed light on a complex network that extends from host metabolism and tRNA modification to viral translational regulation and finally to virion production. We find that 2-thiouridine hypomodification of tRNALys/Glu/Gln causes increased translational frameshifting, changing the ratio of the critical lambda phage proteins gpG and gpGT. We also show that IscU is linked to gpG and gpGT expression by competitive inhibition of the TUS pathway. Knocking out core members of the ISC pathway increases lambda phage replication by relieving competition for sulfur, allowing the TUS pathway to increase the rates of 2-thiouridine formation, thus reducing frameshifting. Targeting tRNA modifications to alter frameshifting rates, to which viral structural proteins are likely to be uniquely sensitive, presents a novel antiviral strategy.\n",
            "In a previous genome-wide screen (Maynard et al, 2010), we found that plates growing TUS pathway deletion strains (ΔtusA, ΔtusE, and ΔmnmA) infected with lambda phage produced plaques of an unusually small diameter compared with wild-type (WT) E. coli. In contrast, several ISC pathway deletion strains (ΔiscU, ΔhscA, and ΔhscB) produced abnormally large diameter plaques compared with WT.\n",
            "We investigated viral replication in these strains more thoroughly by culturing each strain in liquid media, in the presence and absence of virus. Without phage, WT E. coli grew in exponential phase for several hours, slowing and reaching stationary phase as available nutrients decreased (Figure 2A, blue line). In the presence of lambda phage, the culture exhibited three phases (gray line). First, the culture underwent exponential growth similar to the uninfected culture. After ∼6 h in the WT strain, viral lysis overtook the culture and the culture began to clear (lytic phase). Lambda phage is a temperate phage with both lytic and lysogenic life cycles (Landy and Ross, 1977); as a result, after a roughly 3-h lytic phase, lysogenized bacterial population growth took over the culture.\n",
            "The time courses of lambda phage infection of the tusA and iscU knockout strains (the first members of the TUS and ISC pathways, respectively) differed significantly from that of WT. The ΔiscU culture entered the lytic phase very quickly (red curve, Figure 2B), while the infected ΔtusA strain grew in exponential phase substantially longer than the WT strain, leading to a higher turbidity before the lytic phase occurred (green curve). We therefore decided to classify strains that entered the lytic phase earlier as more susceptible to viral infection (red) and those that entered the lytic phase later as less susceptible (green). This susceptibility classification was consistent with our earlier plaque assay experiments (Maynard et al, 2010).\n",
            "We then defined a metric to compare the resistance of different strains with infection with phage lambda. One useful way to summarize the information contained in a time course is to consider each experiment as a vector in n-dimensional space, where n is the number of time points taken. In earlier work, we found that it was useful to normalize these infection time course experiment vectors by the growth rate (Maynard et al, 2010). However, we have since found that bacterial growth and viral growth are tightly linked, and our attempts to numerically correct for or eliminate the effect of bacterial growth rate often distorted the raw data. Bacterial growth is therefore an important (but not sufficient) consideration in classifying our strains.\n",
            "As a result, we determined our new metric simply by calculating the Euclidean distance between the knockout and WT strain time course vectors under infection conditions. Combining this distance with our susceptibility classification, we obtain a displacement vector that points in a negative direction for strains that are less resistant to virus and a positive direction for more resistant strains. The displacement vector, which we call the ‘host resistance', enables us to compare the magnitude and direction of each strain relative with the WT strain and with each other. For example, the ΔiscU strain is significantly less resistant to viral infection than the WT strain, while the ΔtusA strain is more resistant (Figure 2C).\n",
            "As both the TUS and ISC pathways are linked to sulfur metabolism through IscS, we wondered about the interplay between these two pathways and what might be the mechanistic basis for their effect on lambda replication. We began our investigation by examining the impact of other known ISC and TUS pathway members on lambda phage replication (Figure 3A; see Supplementary Figure 1 for line plots of individual replicates). First, we investigated the effect on lambda phage replication of several genes associated with Fe-S cluster biosynthesis. We performed infection time courses of ΔiscU and other strains deficient in members of the iscRSUA-hscBA-fdx-iscX operon (hscA, hscB, iscA, iscR, fdx, and iscX). The slow growth of the iscS deletion strain made it impossible to assess infection dynamics and was not included in the analysis. The ΔiscU, ΔhscA, and ΔhscB strains all exhibited very similar infection dynamics, clearing the culture significantly before WT (Figure 3A, top row; Figure 3B). IscU is the primary scaffolding protein used for Fe-S cluster biosynthesis (Urbina et al, 2001), and HscA and HscB are accessory proteins that help transfer Fe-S clusters to apoproteins (Bonomi et al, 2008) The ΔiscA strain exhibited increased replication efficiency, but less than the hscA, hscB, and iscU gene deletions (Figure 3A and B). IscA assists in Fe-S cluster biosynthesis by shuttling iron to IscU (Yang et al, 2006).\n",
            "Deletion strains for the remaining three genes in the operon, iscR, fdx, and iscX, displayed little if any change in viral infection dynamics from WT (Figure 3A and B). The roles of Fdx and IscX in Fe-S cluster biosynthesis are not known (Takahashi and Nakamura, 1999), but IscR is the transcriptional repressor of the ISC operon and its active form includes an Fe-S cluster (Schwartz et al, 2001). Interestingly, lambda phage replication is slightly but significantly hindered in the iscR knockout strain (Figure 3A and B). Taken together, these observations suggest that ISC pathway members IscU, HscA, HscB, IscA, and possibly IscR have a more significant role in lambda phage infection than Fdx or IscX, and that their WT function helps to restrict phage growth.\n",
            "We next determined how deletion of tRNA thiolation enzymes affected viral susceptibility of the host. We began with the Fe-S cluster-containing enzymes encoded by miaB and ttcA. MiaB is responsible for methylthio formation of ms2i6A at position 37 in tRNAPhe (Pierrel et al, 2004); a sulfur donor for MiaB has not been identified. TtcA catalyzes the 2-thiocytidine modification at position 32 of tRNAArg(ICG), tRNAArg(CCG), tRNAArg(mnm5UCU), and tRNASer(CGU) (Jager et al, 2004). ΔttcA strains displayed doubling times equivalent to WT E. coli (Figure 3A, bottom), and little is known about the functional consequences of modification. Strains lacking either of these genes showed no effect on lambda infection dynamics (Figure 3A, bottom; Figure 3C). In contrast, deletion strains for the two Fe-S cluster-independent tRNA thiolation enzymes ThiI and MnmA displayed a significant increase in viral resistance versus the WT strain (Figure 3A, bottom; Figure 3C). Sulfur is transferred to ThiI through direct binding of IscS (Kambampati and Lauhon, 2000), and ThiI then transfers the sulfur to s4U8 of tRNAPhe. Lambda replication in the ΔthiI strain was mildly but significantly inhibited (Figure 3A, bottom; Figure 3C).\n",
            "The ΔmnmA strain exhibited the most dramatic effect on viral infection of all the strains associated with tRNA thiolation. MnmA catalyzes 2-thiouridination at position 34 of several tRNAs (Kambampati and Lauhon, 2003). Sulfur is supplied to MnmA through the TUS pathway, beginning with the direct binding of IscS by TusA (Ikeuchi et al, 2006). The ΔmnmA strain underwent infection dynamics (Figure 3A, bottom; Figure 3C) very similar to the upstream TUS pathway gene deletion strains we characterized in a previous study (Maynard et al, 2010), including ΔtusA, ΔtusBCD, and ΔtusE. All members of the TUS pathway had a strong, significant, and similar effect on lambda replication (Figure 3A and C), leading us to conclude that Fe-S cluster-independent tRNA thiolation in E. coli exerts a significant effect on lambda phage replication through the TUS pathway and, to a lesser degree, ThiI.\n",
            "Based on the preceding data and a previous observation of increased frameshifting in ΔmnmA strains (Urbonavicius et al, 2001), we hypothesized that reduced lambda replication in the TUS knockouts was due to increased frameshifting during lambda protein synthesis as a result of the decreased tRNA modification. To test this hypothesis, we generated infection curves for several deletion mutants that have been shown to affect frameshifting by loss of tRNA modification, including tgt, truA, miaA, thiI, and mnmE (Urbonavicius et al, 2001). We infected deletion strains for these genes with lambda phage and determined their susceptibility to infection (Figure 4; see Supplementary Figure 2 for line plots of individual replicates). Δtgt, ΔmiaA, and ΔtruA exhibited no significant affect on lambda phage replication, suggesting that the tRNA modifications from these enzymes have no role in lambda phage replication. The remaining knockout strains displayed significant changes in lambda phage's ability to infect the host, including ΔthiI, ΔmnmA (Figures 3 and 4), and ΔmnmE (Figure 4). Intriguingly, MnmE performs a methylaminomethyl modification on carbon 5 of the same tRNA species and nucleoside targeted by MnmA (Elseviers et al, 1984), suggesting that the antiviral properties of the TUS pathway knockouts may be due to specific tRNA modifications instead of to general translational fidelity issues.\n",
            "Host-preferred codon usage is common among many viruses and their hosts (Kunisawa et al, 1998; Lucks et al, 2008). We wondered whether the lambda genome used the codons whose tRNAs are modified by the mnmA, and mnmE gene products preferentially with respect to E. coli. The tRNA species modified by MnmA and MnmE decode the lysine, glutamic acid, and glutamine codons (Kambampati and Lauhon, 2003); we examined the codon usage for lysine, glutamic acid, and glutamine in lambda phage and in E. coli for synonymous codon usage bias. Lambda phage encodes lysine 748 times (65% AAA and 35% AAG), glutamic acid 900 times (57% GAA and 43% GAG), and glutamine 576 times (20% CAA and 80% CAG), with similar biases in E. coli (Table II). This similarity in codon usage does not support the hypothesis that lambda phage protein synthesis is any more impaired in strains harboring hypomodified 2-thiouridine tRNAs than its host.\n",
            "Since we thought it unlikely that lambda phage replication was being significantly affected by general reading frame maintenance in the TUS pathway knockout mutants, we turned instead to specific tRNA modifications that could explain our observations of an increase in viral resistance. In particular, our attention was drawn to PRF as a specific instance of frameshifting that could be particularly sensitive to changes in frameshifting frequency.\n",
            "In lambda phage, PRF occurs during translation of the lambda phage GT region, which encodes two possible protein products, either gpG or gpGT; expression of gpGT depends on ribosomal frameshifting at the slippery sequence GGGTTTG (Levin et al, 1993). This sequence encodes the dipeptide Gly-Lys in both the 0 open reading frame (ORF; GGT-TTG) and the −1 ORF (GGG-TTT). In the 0 ORF, which is maintained ∼96% of the time, gpG is expressed (Figure 5A); and in the remaining cases, the ribosomal machinery reaches the slippery sequence and slips back one base pair. As a result, the stop codon for gpG, immediately downstream of the slippery sequence, is bypassed and the larger gpGT product is produced. Interestingly, both the lysine codons TTG (0 ORF) and TTT (−1 ORF) are decoded by the same tRNALys (UUU), a target of MnmA and MnmE. We therefore wondered whether TUS pathway gene deletion strains would exhibit an altered rate of frameshifting and consequently a change in the ratio of gpG to gpGT.\n",
            "To monitor gpG and gpGT expression in E. coli, we created a plasmid for the inducible synthesis of these proteins (pBAD-λGT; Figure 5B). We PCR amplified the GT region of the lambda genome and introduced this region behind an arabinose-inducible promoter, to control expression of the transcript, and the Xpress Epitope C-terminal tag, to facilitate detection by immunoblotting. We then transformed WT, ΔtusA, ΔiscU, and ΔtusAΔiscU strains with pBAD-λGT. Deletion of the iscU gene led to abrogation of frameshifitng, whereas frameshifting frequency increased in both the ΔtusA and the ΔtusAΔiscU strains relative to WT (Figure 5C). We also tested frameshifting frequency for additional strains (ΔmnmA, ΔthiI, ΔmiaB, ΔttcA, ΔhscA, ΔhscB, Δfdx, ΔiscA, and ΔiscR) and found frameshifting levels to be consistent with infection dynamics (see Supplementary Figure 3). We interpret these results to signify that the decreased thiolation of tRNALys (UUU) in the TUS pathway knockout strains leads to an increase in frameshifting at the slippery sequence.\n",
            "Surprisingly, the ΔiscU strain exhibited markedly reduced gpGT levels relative to the WT strain (Figure 5C), strongly indicating that both IscU and TusA act on lambda phage replication by altering the gpG:gpGT ratio. Furthermore, our observation that the miaB and ttcA deletion strains exhibited normal lambda phage replication (Figure 3) suggested that IscU does not act directly through thiolation of specific tRNA modifications. How, then, does IscU influence viral replication independent of direct tRNA modification?\n",
            "One possible answer is that IscU competes with TusA to bind IscS. Recent X-ray crystallography studies showed that IscU and TusA bind IscS at distinct but adjacent locations (Shi et al, 2010). Superposition of the IscU and TusA structures indicated a spatial overlap between the volume occupied by TusA and IscU when bound to IscS. Furthermore, three-way pull-down experiments demonstrated that IscS may bind either TusA or IscU individually, but not both at the same time (Shi et al, 2010). To further explore the possibility that thiolation of tRNALys (UUU) is affected by competition between TusA and IscU for sulfur transfer from IscS, we constructed a mathematical model of IscU and TusA binding to IscS (Figure 6A). Our model consisted of a set of equilibrium and mass conservation relationships. In this case, the model parameters had either been previously measured or were not difficult to estimate based on existing data (Urbina et al, 2001; Taniguchi et al, 2010). Our model predicts the relative amount of U34 thiolation from the TUS pathway for a specified amount of IscU.\n",
            "We combined this protein binding model with our previous mathematical model of E. coli infection by lambda phage (Maynard et al, 2010). Our earlier model predicts the E. coli time course dynamics of lambda phage infection based on three parameters: the fraction of infections that are lytic instead of lysogenic, the rate of infection, and the burst size (how many functional progeny are released upon lysis). To integrate these two models, we related U34 thiolation (the output of the protein binding model) to the burst size (input for our infection model). We were thus able to explore the effect of changing the IscU or TusA concentration on the production of functional phage and the resulting infection time course. We first tested the integrated model's ability to predict the outcome of deleting iscU or tusA, and in both cases the model was able to capture the infection time course (Figure 6B–E; see Supplementary Figure 4 for line plots of individual replicates). We also considered titration of IscU experimentally as well as computationally by introducing an IPTG-inducible iscU gene construct into a ΔiscU strain and determining the infection time courses under various IPTG concentrations. The effect of IscU titration was predicted well by our competitive inhibition model (Figure 6F and G; see Supplementary Figure 4 for line plots of individual replicates).\n",
            "Finally, we considered the case of a double deletion of tusA and iscU. The competitive inhibition model suggested that the double mutant would behave in a manner similar to a ΔtusA strain under infection conditions (Figure 6H). Again, this prediction agreed with the experimental time course (Figure 6I and J; see Supplementary Figure 4 for line plots of individual replicates), as well as our experimental observation that the ratio of gpGT to gpG in the double mutant was similar to that observed in the ΔtusA strain, but not in the ΔiscU strain (Figure 5C).\n",
            "As a control, we compared the predictions of the competitive inhibition model with those of the alternative hypothesis that IscU and TusA influence viral infection independently. This hypothesis required some relatively simple modifications in the equations and parameters of our competitive binding model (Materials and methods). Our independent effect model was able to capture the results of tusA deletion, iscU deletion, and IscU titration as accurately as the competitive inhibition model (data not shown). However, the independent effect model was not able to predict the behavior of the double knockout strain (Figure 6K).\n",
            "Our competitive inhibition model predicts that the removal of IscU, one of TusA's primary competitors for sulfur, will cause an increase in the fraction of 2-thiolation modified U34 tRNALys(UUU) relative to BW25113. To directly measure the fraction of thiolated tRNALys(UUU), we performed an [(N-Acryloylamino)phenyl] mercuric chloride (APM) northern blot (Figure 7), which specifically retards the migration of the thiolated tRNA fraction (Igloi, 1988). We found a small fraction of hypomodified tRNALys in BW25113, and as predicted by the competitive inhibition model, loss of IscU in the ΔiscU strain decreased this fraction even further. We therefore conclude that our competitive inhibition model, but not our independent effect model, is sufficient to explain the effect of TUS and ISC pathway knockouts on lambda phage replication.\n",
            "Taken together, our observations suggest the existence of a complex network that links sulfur metabolism, tRNA modification, competitive binding, and PRF-based regulation of viral protein expression with host susceptibility to viral infection (Figure 8). TusA obtains sulfur from the cysteine desulfurase IscS, and passes it along the TUS pathway for eventual modification of tRNALys(UUU) at U34. The fraction of tRNALys(UUU) that undergoes this modification influences the frequency of PRF in lambda phage due to its role in decoding part of the slippery sequence within its GT region. Although lambda phage requires both gpG and gpGT for normal replication, the gpG:gpGT ratio must be kept high. Deletion of any member of the TUS pathway prevents modification of tRNALys(UUU), and subsequently increases frameshifting, both decreasing the gpG:gpGT ratio and lambda phage production.\n",
            "In the presence of HscA, HscB, and (less critically) IscA, IscU binds IscS more strongly than TusA (Shi et al, 2010) and is therefore able to outcompete TusA for sulfur. When IscU, HscA, HscB, or IscA is removed from the system, TusA obtains more sulfur than normal, leading to hypermodification of tRNALys(UUU), a decrease in frameshifting, and increases in the gpG:gpGT ratio and lambda phage production. Previous investigations in B. subtilis reported that disrupting the function of IscU affects several levels of tRNA modifications, including an increase in 2-thiolation of U34 (Leipuviene et al, 2004); our observations suggest that in the case of tRNALys(UUU), this effect is a result of an increase of sulfur flux through the TUS pathway (Figures 5C, 6K, and 7). Removal of the transcriptional repressor IscR leads to an increase in protein production from the ISC operon (Schwartz et al, 2001), reducing the amount of sulfur available to TusA. This competition effect depends on an intact TUS pathway, which is why the ΔtusAΔiscU deletion mutant expresses a phenotype similar to the ΔtusA strain.\n",
            "This model fits our observations well, but unresolved issues remain, most notably the role of ThiI in viral infection. ThiI would seem to be another possible competitor for sulfur from IscS (Figure 2A). However, the ΔthiI strain exhibited decreased lambda replication, albeit somewhat less strongly than the TUS pathway deletion strains. One clue to this mystery may reside with the CyaY protein, which is thought to be an iron donor for Fe-S cluster assembly, forming a ternary complex with IscU and IscS (Shi et al, 2010). The binding site for ThiI on IscS overlaps substantially with CyaY (Shi et al, 2010), and it is conceivable that in the absence of ThiI, CyaY is able to interact more efficiently with IscU, either stabilizing the IscS-IscU-CyaY complex or assisting sulfur transfer to IscU. In either case, deleting thiI may increase IscU's ability to compete with TusA for sulfur.\n",
            "Our findings point to several novel antiviral strategies and targets. Although this work was performed using E. coli and phage lambda, many of the key elements of the network we describe are conserved in other systems, in both in host and the virus. Both Fe-S clusters and tRNA modifications are ubiquitous in cells (Limbach et al, 1994; Py and Barras, 2010) and PRF is a common viral translational strategy (Gesteland and Atkins, 1996). Indeed, frameshifting has already been identified as a potential target in HIV, but efforts to target the gag-pol mRNA secondary structure have been challenging (Hung et al, 1998). Our results point to upstream (e.g., tRNA modifying enzymes) and even indirect targets (e.g., Fe-S cluster biosynthesis pathways), which can just as potently alter infection and in some cases are tunable (Figure 6G). We therefore anticipate that our observations and inferences may have broad relevance to understanding other infections, including viral infection of humans.\n",
            "The E. coli strains used in this study were all taken or derived from the Keio Collection, a comprehensive library of single gene deletion strains (Baba et al, 2006). BW25113 is the parental strain used for the construction of the Keio Collection. Lambda phage was obtained from the ATCC (23724-B2). The ΔtusBCD strain was generated previously (Maynard et al, 2010), and the ΔtusAΔiscU double knockout strain was constructed according to the Datsenko-Wanner method (Datsenko and Wanner, 2000). Primer sequences are available upon request. The lambda GT reporter strain was constructed using a pBAD/His A plasmid (Invitrogen, Cat# V430-01). The lambda phage GT region was amplified via PCR (primers: 5′-cacattctcgagatgttcctgaaaaccgaatca-3′ and 5′-gtgtaaaagcttgccataccggactcctcct-3′). The amplified GT region and pBAD/His A plasmid were cut with XhoI and HindIII. BW25113, ΔtusA, ΔiscU, and ΔtusAΔiscU strains were transformed with the arabinose-inducible lambda phage GT reporter construct (Figure 5B). A chemically inducible ΔiscU strain was created by transforming ΔiscU with pCA24N-iscU isolated from the ASKA collection (Kitagawa et al, 2005).\n",
            "We monitored infection dynamics using an incubated plate reader (Perkin-Elmer Victor3, 2030-0030) under growth conditions described previously (Maynard et al, 2010). Exponentially growing cells were normalized to 0.1 OD 600 nm. Fifteen microliters of 0.1 OD E. coli and 15 μl of ∼104 plaque forming units (p.f.u.)/ml lambda stock were added to 170 μl of LB medium in 96-well plates, representing a multiplicity of infection of ∼2 × 10−4 p.f.u./bacteria. Each plate contained four technical replicates of infected and uninfected samples. The plate reader measured the absorbance of the culture approximately every 17 min for 16 h. Each plate contained a sample of BW25113 for between-plate normalization. Three independent biological replicates were performed for each strain.\n",
            "Time courses were compared based on a useful displacement metric that includes a direction in addition to a distance. The distance was determined by calculating the Euclidean distance between the two time course vectors; the Euclidean distance between the recorded absorbances for each culture were calculated at each time point, and these distances were summed for all time points. For our purposes, the direction could only hold two values, +/−1. The positive value was associated with strains exhibiting increased resistance to viral infection, and the negative value was associated with increased susceptibility to viral infection. The direction was determined by the difference in the time it took a culture to reach its peak absorbance. For example, time courses with longer times to peak compared with BW25113 were considered to be displaced in a positive direction from BW25113. The displacement was then calculated as (direction) × (distance).\n",
            "The Euclidean distance from BW25113 was measured for each biological replicate. Statistical significance was determined using a two-sample t-test comparing the Euclidian distances calculated for each biological replicate and its distance from BW25113 for each plate containing the strain being compared.\n",
            "Lambda phage ORFs were extracted from NCBI Reference Sequence NC_001416.1 (Enterobacteria phage lambda, complete genome) and codon usage was tallied.\n",
            "Codon usage for WT E. coli (W3110) was determined from the Codon Usage Database (http://www.kazusa.or.jp/codon/).\n",
            "pBAD-λGT-transformed cells were cultured overnight at 37°C in LB and the appropriate antiobiotics. The cultures were diluted 1:50 into fresh LB the following morning and placed back in the shaker for 1 h. The culture was then split into 1 ml aliquots arabinose. Arabinose induction was performed for 2 h at 0.2, 0.02, and 0.002% arabinose, after which the samples were spun down in a table-top centrifuge at ∼20 000 g for 30 s. The media was aspirated and the pellets were frozen at −80°C. The samples were run on a 12% SDS–PAGE gel for 1.5 h at 80 V in Tris/glycine/SDS buffer. The protein was transferred onto a polyvinylidene fluoride membrane for 1 h at 100 V at room temperature. The membrane was washed for 30 s in TBST, then blocked with 5% milk blocking buffer for 1 h. The primary antibody (Anti-express antibody, Invitrogen) was diluted 1:5000 and incubated with the membrane blot overnight at 4°C. The membrane was washed five times for 10 min with TBST, followed by a 1-h incubation with a 1:5000 dilution of the secondary antibody (goat anti-mouse IgG horseradish peroxidase). The membrane was washed five times for 5 min with TBST. Luminol reagent was used for chemiluminescence imaging (Santa Cruz Biotechnology, sc2048).\n",
            "BW25113 and ΔiscU were cultured overnight and diluted 1:100 to 20 ml in the morning. When the cultures reached ∼0.5 OD 600 nm, the samples were placed on ice. The cells were pelleted at 5000 r.p.m. for 5 min at 4°C and the supernatant was decanted. tRNA enrichment was performed using standard techniques. Briefly, the cell pellets were resuspended in 300 μl cold sodium acetate (0.3 M, pH 4.5, 10 mM EDTA) and transferred to eppendorf tubes. The samples were snap frozen in liquid nitrogen. After thawing, 300 μl phenol:chloroform (pH 4.7) was added to the samples and vortexed four times for 30 s bursts with 1 min incubation on ice in between. The samples were centrifuged for 15 min at 4°C in a table-top centrifuge at maximum speed. The top layer (aqueous phase) was transferred to a new tube and the acid phenol extraction procedure was repeated. After the aqueous phase was transferred to a new tube, the RNA was ethanol precipitated by adding three volumes of cold 100% ethanol and spinning at maximum speed for 25 min at 4°C. The RNA pellet was resuspended in 60 μl cold sodium acetate (0.3 M, pH 4.5) and precipitated again by adding 400 μl 100% ethanol and spinning at maximum speed for 25 min at 4°C. The supernatant was decanted and all traces of ethanol were removed. The RNA pellet was air dried on ice then resuspended in 100 μl cold sodium acetate (10 mM, pH 4.5).\n",
            "The APM northern blot was performed according to the protocol specified by Igloi (1988). Briefly, two 8 M Urea, 5% acrylamide gels were cast in 0.5 × TBE buffer. One gel contained 200 μl of 1 mg/ml APM (kindly supplied by Gabor Igloi) for 10 ml of gel. The gels were prerun for 30 min at 180 V in 0.5 × TBE. Approximately 2.5 μg of RNA (for each well) was then mixed with 2 × loading buffer and heated to 65°C for 3 min before being cooled on ice for 1 min. The gels were run for ∼45 min at 180 V. The RNA was transferred onto a nylon membrane using a semi-dry system for 1 h at 250 mA. The transfer buffer contained 0.5 × TBE supplemented with 10 mM β-mercaptoethanol. The RNA was then UV crosslinked to the nylon membrane. T4 Polynucleotide Kinase (PNK) was used to radiolabel 25 pmol of 32P ATP to 25 pmol of tRNALys(UUU) oligonucleotide probe for each blot. The oligonucleotide sequence used to probe tRNALys(UUU) was 5′-TGGGTCGTGCAGGATTCGAA-3′. Qiagen's Qiaquick Nucleotide Exchange Kit was used to remove free 32P ATP. The blots were incubated in 25 ml of prehybridization buffer (10 × Denhardt's, 6 × SSC, 0.1% SDS) at 42°C for 1 h. The labeled probe was heated to 95°C for 2–3 min before being added to the prehybridaztion buffer. The blots were hybridized at 42°C overnight. The next day, the blots were washed with high salt buffer (5 × SSC, 0.1% SDS) at 37°C for 20 min twice, followed by two washes with low salt buffer (1 × SSC, 0.1% SDS) at 42°C for 20 min. After the final wash, the blots were exposed to a phosphorscreen for 10 min.\n",
            "We constructed a competitive inhibition model based on the following assumed mechanisms: (1) the cysteine desulfurase IscS binds sulfur, converting cysteine to alanine in the process; (2) thiolated IscS presents sulfur to the apo forms of TusA and IscU; (3) TusA and IscU bind thiolated IscS with their respective affinities and obtain the sulfur; (4) TusA and IscU transfer the sulfur to their downstream pathways, represented as sTUS and sISC, respectively. The source code for our model can be downloaded from https://simtk.org/home/lambda-tus.\n",
            "The reactions are as follows:\n",
            "\n",
            "From these reactions we derived the following set of ordinary differential equations and mass conservation relationships:\n",
            "\n",
            "Many of the parameter values were taken from the literature. Taniguchi et al (2010) reported values of ∼800 IscS proteins per cell and 450 IscU proteins per cell. Approximating the E. coli volume as 10−15 L, we obtained concentrations of 1.37 and 0.77 μM for IscS and IscU, respectively. Since TusA amounts were not reported, we estimated its concentration to be similar to but less than IscU 0.5 μM. An in-vitro study reported the Kd values of IscS and IscU to be 2 μM, and the kcat (k2 in the above equations) to be 8.5 per minute (Urbina et al, 2001). IscU displaced TusA in a three-way pull-down assay (Shi et al, 2010), from which we hypothesized that the Kd of TusA-IscU was 20% more than that of IscU-IscS. The cysteine concentration was chosen to be equal to the IscS concentration.\n",
            "The forward rates of reaction were assigned as follows: k1=k3=k5=k7=k8=105, k4=k6=k−3. The reverse reaction rates were obtained by multiplying the corresponding forward rate by the appropriate dissociation constant. Gene deletion conditions were simulated by setting the pertinent concentration values to zero. IscU titrations were modeled by setting its concentration to 50, 100, and 150% of the WT amount.\n",
            "Steady-state values of the sTUS parameter relative to WT were used to scale the amplification factor b in our previous non-dimensionalized predator-prey model (Maynard et al, 2010). We simulated infection dynamics with the following parameters: , μ*=0.3, K*=0.4, ks=0.\n",
            "The independent effect model hypothesizes that there is no competition between IscU and TusA for sulfur. Thus, they each receive their own (equally sized) pool of thiolated IscS (iscSU·S belongs to IscU and iscSA·S belongs to TusA). The dissociation constants of both IscU and TusA with their respective pools of IscS were set to 2.7 μM. The concentrations of IscU and TusA were set to the same values as in the competitive model. The amount of thiolated IscS, however, was increased 100-fold to prevent it from being limiting. The reactions are as follows:\n",
            "\n",
            "From these, we derived the following set of ordinary differential equations and constraints:\n",
            "\n",
            "The kinetic parameters were specified as follows: k1f=k2f=k3f=k4f=105. Non-zero reverse rates were set to the product of the forward rate and the appropriate dissociation constant.\n",
            "Steady-state values of the sTUS and sISC parameters relative to WT were used to scale the amplification factor b. Parameters for the predator-prey equation were . We see that in both models, the WT burst rate is 20 and the tusA knockout burst rate is 10. Supplementary Table 1 describes the parameters used in the two models.\n",
            "The standard ode23s solver in Matlab was used for solving the sulfur transfer equations. The ode45 solver was used to solve the predator-prey equations.\n",
            "***************************************************************************************\n",
            "['Emerging and re-emerging viral infections are becoming severe global public health problems in the current century. In February 2003, an outbreak of severe acute respiratory syndrome (SARS), caused by SARS coronavirus (SARS-CoV), was reported in Guangdong, China (Ksiazek et al. 2003). The SARS-CoV infected 8096 cases and 774 deaths worldwide. In March 2009 a novel influenza virus (H1N1pdm) emerged in the United States and Mexico. H1N1pdm obtained the capacity to transmit in humans and quickly spread to more than 214 countries (Hendrickson and Matthay 2013). Thereafter the H1N1 infection became a seasonal virus circulating over the world (Dawood et al. 2009). Middle East respiratory syndrome (MERS) coronavirus was isolated from patients who developed acute pneumonia and renal failure in 2012 from Saudi Arabia (Zaki et al. 2012). One big secondary outbreak with 186 confirmed cases was in South Korea in 2015. Up to January 2020, more than 2500 confirmed cases were reported with a case fatality of 34.4% (WHO 2019). In December 2019, a novel coronavirus, SARS-CoV-2 caused an outbreak of acute pneumonia in Wuhan City of China (Zhou P et al. 2020).The current animal models for study of SARS-CoVs include African green monkeys, macaques, ferret, and mice (Kong et al. 2009; Smits et al. 2011). These in vivo models are useful for studying many important questions, for example, local or systemic pathogenic changes, immune response, and drug metabolism and in vivo efficacy. However, these animal models have limitations due to species difference. The viral tropism correlates to expression of viral receptor(s) on the surface of host cells and intracellular restriction as well. Cancer cell lines have been used widely for biological or functional cell systems in many other fields (Agarwal and Rimm 2012; Barretina et al. 2012; Palechor-Ceron et al. 2019). These cells are defective for many functions of normal human cells because of significant genetic and epigenetic changes compared to the normal tissues. However, primary cell yields and viability from any original tissue specimens are often relatively low or highly variable. These normal cells cease at a few passages because of very limited proliferative capacity in vitro. It has been extremely difficult for decades to generate and maintain normal cell lines. Several exogenous immortalization approaches have been used to establish primary cell lines from variety of tissue types. For example, viral oncogenes such as SV40 large T antigen or the human papillomavirus (HPV) E6/E7 proteins can bypass the cell senescence block through interfering with p53 and Rb regulatory pathways (Liu et al. 2005; Liu et al. 2007; Liu et al. 2008; Cid Arregui et al. 2012; Klingelhutz and Roman 2012; Ghittoni et al. 2015). Expression of exogenous cellular genes (hTERT, c-Myc, cdk4, etc.) or inactivation of cellular tumor repressors (p53, pRB, etc.) can be widely used to immortalize primary human normal cells, which leads to disrupted cell differentiation or loss of tissue type-specific functions (Liu et al. 2012).', 'We observed that two classes of functions are required for cell immortalization by HPV E6 and E7: telomerase activation and actin cytoskeleton alterations (Fu et al. 2003; Charette and McCance 2007; Liu et al. 2007; Liu et al. 2008; Yue et al. 2011; Klingelhutz and Roman 2012). We also noted that feeder cells and altered molecular pathways activate telomerase (Fu et al. 2003; Liu et al. 2008; Liu et al. 2009; Klingelhutz and Roman 2012; Liu et al. 2012), and a Rho-Kinase inhibitor, Y-27632, disrupts the actin cytoskeleton and inactivate Rho (Chapman et al. 2010; Liu et al. 2012). Unexpectedly, combination of feeder layers and Y-27632 allows to rapidly establish both normal and tumor cell cultures from non-keratinocyte tissues (Liu et al. 2012; Suprynowicz et al. 2012; Palechor-Ceron et al. 2013; Liu et al. 2017). We also demonstrated that the effect of the combined co-culture condition is rapid (2 days), and the whole cell populations are shifted or reprogrammed rather than a clone selection as conventional cancer cell lines (Liu et al. 2012; Suprynowicz et al. 2012). These cell cultures stop proliferating if one of the conditions (Y-27632 or feeder layer) is not met. We termed this culture method and resulting cells as “conditional reprogramming (CR)” and “conditionally reprogrammed cells (CRC)”, respectively (Liu et al. 2012; Suprynowicz et al. 2012; Palechor-Ceron et al. 2013; Liu et al. 2017; Palechor-Ceron et al. 2019). Indeed, organoids (Weeber et al. 2017; Puca et al. 2018; Sachs et al. 2018; Mullenders et al. 2019; Xinaris 2019) and CR technologies have been widely used in cancer biology and regeneration fields. They both have been recognized as the key new technologies by NCI (National Cancer Institute) precision oncology (https://ocg.cancer.gov/programs/hcmi/research) (Friedman et al. 2015; Senft et al. 2017), which are used for HCMI (human cancer model initiatives) program launched during 2019 annual AACR) (https://www.atcc.org/en/Products/Cells_and_Microorganisms/HCMI.aspx?utm_id&equals;t18020438l1). Table 1 lists properties of these cell models. In this review, we will highlight roles of CRC and coupled air-liquid interface (ALI) system in studies of emerging viruses.', 'Primary airway epithelial cells can be obtained from airway tract using non-invasive or minimally invasive techniques, nasal brushing, nasorophynx swaps, induced sputum sample collection, bronchiolar lavage, endobronchial brushing or biopsy. After CR technology was established in 2012, the CR protocol has been proven to easily establish patient-derived normal and cancer cell cultures without genetic manipulation (Liu et al. 2012; Palechor-Ceron et al. 2013; Liu et al. 2017). Originally, the CR used irradiated mouse fibroblast cells (swiss mouse 3T3, J2 clone) and the Rho-associated kinase inhibitor (Y-27632) to propagate epithelial cells (Liu et al. 2012; Palechor-Ceron et al. 2013; Liu et al. 2017). A few improvements have been used to simplify protocols using J2 conditioned medium (Liu et al. 2012; Palechor-Ceron et al. 2013; Liu et al. 2017), hypoxia condition (1%–2% O2) (Peters-Hall et al. 2018), and combination with dual TGF-beta/SMAD inhibition (Mou et al. 2016; Zhang C et al. 2018). CR technology is simple and cheap since there is no need for expensive reagents as matrigel for organoids, robust since 1 × 106 cells can be generated from a needle biopsy within 7 days, and rapid since the whole populations of cells can be reprogrammed within 2 days instead of clonal selection. Figure 1 shows a diagram of normal cell cultures in CR conditions and 3D (ALI and non-iPS organoids) conditions for ex vivo models of airway epithelial cells.', 'CR allows the rapid generation of multi-lineage cell cultures (conditionally reprogrammed cells, CRC) from many different normal epithelial tissues, as well as tumors derived from these sites (Liu et al. 2012; Suprynowicz et al. 2012; Palechor-Ceron et al. 2013; Liu et al. 2017; Palechor-Ceron et al. 2019). Normal cell lines retain their normal karyotype and growth properties, and tumor cells retain their tumorigenic potential (Liu et al. 2012; Suprynowicz et al. 2012; Palechor-Ceron et al. 2013; Borodovsky et al. 2017; Liu et al. 2017; Yuan et al. 2017; Alkhilaiwi et al. 2018; Correa et al. 2018; Palechor-Ceron et al. 2019). Interestingly, generation of CRC from normal tissue is reversible, and manipulating CR conditions allows the cells to differentiate normally (Suprynowicz et al. 2012; Liu et al. 2017). For example, when CRC from skin epithelium or tracheal epithelium are placed in an ALI culture system, the skin cells form a well-differentiated stratified squamous epithelium, whereas the tracheal cells form a ciliated airway epithelium.', 'ALI (Air–Liquid interface) culture (Fig. 2) of CR airway cells is uniquely suited as a model for in vitro infection studies. This is due to the ability of ALI cultures to faithfully recapitulate key characteristics of the in vivo airway. For example, ALI cultures exhibit relevant proportions of airway cell types, appropriate cellular polarization and junctional properties, dynamic physiologic processes such as mucus secretion and coordinated ciliary beating, and physiological expression and subcellular localization of characteristic proteins bearing species-specific sequences.', 'To date, CR by itself or in combination with ALI cultures have been used in studies of host defense and viral infections, drug screening and toxicity testing, wound healing/repair, gene therapies as ex vivo preclinical models for lung cancer, chronic obstructive pulmonary disease, cystic fibrosis, and asthma (Yuan et al. 2012; Bove et al. 2014; Crystal et al. 2014; Saenz et al. 2014; Chu et al. 2015b; Feng et al. 2015; Kotha et al. 2015; Walters et al. 2015; Alamri et al. 2016; Beglyarova et al. 2016; Butler et al. 2016; Ellis et al. 2016; Papapetrou 2016; Reynolds et al. 2016; Borodovsky et al. 2017; Gentzsch et al. 2017; Jensen et al. 2017; Li et al. 2017; Mahajan et al. 2017; Martinovich et al. 2017; Wolf et al. 2017; Yu et al. 2017; Yuan et al. 2017; Alkhilaiwi et al. 2018; Boström et al. 2018; Brewington et al. 2018; Correa et al. 2018; Jin et al. 2018; LaRanger et al. 2018; Mimoto et al. 2018; Moorefield et al. 2018; Peters-Hall et al. 2018; Shay et al. 2018; Wang et al. 2018; Zhang Z et al. 2018; Alkhilaiwi et al. 2019; Jiang et al. 2019; Krawczyk et al. 2020; Liu 2019; Martini et al. 2019; Mondal et al. 2019; Nicolas et al. 2019; Palechor-Ceron et al. 2019; Su et al. 2019; Chai et al. 2020). Compared to organoids (close 3D cultures) shown in Fig. 1 (right lower panel), ALI system is much easier for virology studies because of an open apical area for infections or viral inoculations (Fig. 3). Here we will focus our discussion on applications of this physiological system in respiratory virus infections. Figure 3 shows many aspects of studies of emerging viruses.', 'Roberts et al. collected samples from paired nasal and bronchial brushings and cultured and expanded nasal and bronchial epithelial cells using CR condition (irradiated 3T3 fibroblasts in presence of Y-27632) (Roberts et al. 2018). Passaged cells were then put onto collagen coated transwells in 12 well plates for ALI culture. They found that viral infection in both cell types increased the expression of IP-10 (interferon gamma-induced protein 10), although the increase was only significant in the ALI culture, with combining rhinovirus infection and IL-13 treatment (Roberts et al. 2018). Kotha et al. showed that the level of the apical adenovirus receptor (CAREx8) and physiologically relevant levels of IL-8 and neutrophils (components of the innate immune system) enhanced entry of adenovirus in polarized human airway epithelia (Kotha et al. 2015). Jonsdottir et al. established CRC and ALI cultures from both upper and lower airway to study the host innate immune response to human coronavirus 229E (HCoV-229E) and human respiratory syncytial virus (RSV) after gene manipulations (Jonsdottir et al. 2019). It is well known that autonomous parvovirus replication depends on the S phase of the host cells. Interestingly, Deng et al. (2016) reported for the first time parvovirus DNA replicated in non-dividing cells autonomously. They first established CR cells and ALI cultures (non-dividing airway epithelial cells) and inoculated ALI with human parvovirus HBoV1. Their results demonstrated that HBoV1 infection of ALI cultures induces a DNA damage response (DDR), thereby facilitating viral genome amplification. They also discovered that Y- family DNA repair polymerases, Pol η and Pol κ, are involved in HBoV1 genome amplification in ALI system. This is the first report to show that parvovirus DNA replicates in non-dividing cells autonomously. Zhu et al. studied HSV-2 infection in CR and ALI cultured normal vaginal epithelial cells (Zhu et al. 2017). Fink et al. also established CR cells from routine vaginal repair surgeries or hysterectomies and studied antiviral drug (Arbidol) Inhibition of Zika virus (Fink et al. 2018). Finally, Imai-Matsushima et al. generated long-term culture of distal airway epithelial cells and differentiated alveolar epithelial cells that are suited for influenza virus study (Imai-Matsushima et al. 2018).', 'On Jan. 30 and March 11, 2020, the International Health Regulations Emergency Committee World Health Organization (WHO) declared the COVID-19 or SARS-CoV-2 infection outbreak as a “public health emergency of international concern”, and characterized COVID-19 as a pandemic, respectively (WHO 2020). COVID-19 cases have ranged from very mild (including asymptomatic), mild, moderate, severe, and critical severe including illness resulting in death (Lu et al. 2020; Singhal 2020; Wang et al. 2020; Wu and McGoogan 2020; Yang S et al. 2020; Yang X et al. 2020; Ye et al. 2020; Zhang et al. 2020; Zhou C et al. 2020; Zhou F et al. 2020; Zou et al. 2020). COVID-19 common symptoms include fever, cough and shortness of breath. Muscle pain, sputum production and sore throat are less common (Lu et al. 2020; Singhal 2020; Wang et al. 2020; Wu and McGoogan 2020; Yang S et al. 2020; Yang X et al. 2020; Ye et al. 2020; Zhang et al. 2020; Zhou C et al. 2020; Zhou F et al. 2020; Zou et al. 2020). While the majority of cases result in mild symptoms, some progress to severe pneumonia and multi-organ failure including lung, heart, and kidney (Lu et al. 2020; Singhal 2020; Wang et al. 2020; Wu and McGoogan 2020; Yang S et al. 2020; Yang X et al. 2020; Ye et al. 2020; Zhang et al. 2020; Zhou C et al. 2020; Zhou F et al. 2020; Zou et al. 2020). The rate of deaths per number of diagnosed cases is estimated to be 3.4% but varies by age and other health conditions (Lu et al. 2020; Singhal 2020; Wang et al. 2020; Wu and McGoogan 2020; Yang S et al. 2020; Yang X et al. 2020; Ye et al. 2020; Zhang et al. 2020; Zhou C et al. 2020; Zhou F et al. 2020; Zou et al. 2020). The mechanisms how SARS-CoV-2 infects human airway epithelial cells and also causes severe multi-organ failure are largely unknown. Generally speaking, early stage of COVID-19 or cases with very mild (including asymptomatic), mild symptoms are usually due to rapid replication of viruses at local areas, innate response of the host cells and local immune-response (IgA). Severe cases with multi-organ injury are usually due to possible viral replications in the target organs and immunopathogenic injuries (including cytokine storm or antibody dependent enhancement). Recently, two reports demonstrated that large amounts of SARS-CoV-2 were detected in the upper airway and saliva (To et al. 2020; Zou et al. 2020).', 'SARS-CoV-2 and SARS-CoV share the same functional host-cell receptor ACE2 (Hoffmann et al. 2020; Walls et al. 2020; Wan et al. 2020; Wrapp et al. 2020; Zhou P et al. 2020) and sequence analysis reveals that SARS-CoV-2 possesses crucial amino acid residues for ACE2 binding (Hoffmann et al. 2020). ACE2 predominantly expresses in vascular endothelial cells, kidney and heart tissues, small intestine, and testes (Hamming et al. 2004). The distribution of ACE2 in human tissues does not seem to correlate with diseases with COVID-19, while this may help explain why severe cases with COVID-19 have multi-organ failure including heart and kidney. It is important to understand body site-specific or tissue-specific viral replication, innate immune response, and infectivity. Two recent reports demonstrated that large amounts of SARS-CoV-2 were found in the upper airway and saliva as we described above (To et al. 2020; Zou et al. 2020), where ACE2 is not expressed or expresses at very low levels. Wolfel et al. performed a dynamic viral analysis of nine COVID-19 patients, they showed active virus replication in upper airway tissues. Virus shedding at pharynx was very high at the first week of symptoms, the peak was at 7.11 × 108 RNA copies per throat swab at day 4. SARS-CoV-2 viruses were readily cultured from throat- and lung-derived samples, but not from stool samples with high viral RNA. Viral load was declined after 7 days of symptoms (Wolfel et al. 2020). A ferret model of SASR-CoV-2 infection and transmission was established recently (Kim et al. 2020), which can recapitulate some aspects of human COVID-19 disease. SARS-CoV-2-infected ferrets shed virus in nasal washes, saliva, urine, and feces up to 8 days after infection. Viral antigens were also detected in several tissues, as nasal cavity, trachea, lung, and intestine. This ferret model represents an animal model of SARS-CoV-2 infection and transmission, may facilitate to develop COVID-19 therapeutics and vaccine as well. However, distribution of ACE2 and viral replication in different organs or tissues, or possible other receptors or co-receptors in this model need to be further determined. Another study also demonstrated that SARS-CoV-2 may replicate well in ferrets and cats, but poorly in dogs, pigs, chickens, and ducks (Shi et al. 2020). These inconsistences suggest possible alternative receptors or co-receptors on the surface of the target cells in different tissues. A study by Blanco-Melo et al. suggests that human alveolar adenocarcinoma cells (A549) are able to support viral replication of SARS-CoV-2, RSV and influenza A virus, despite undetectable levels of ACE2 and TMPRSS2, the putative receptor and protease for SARS-CoV-2 entry (Blanco-Melo et al. 2020). To solve these problems, we will need comprehensive studies of tissue staining of viral antigens and cellular proteins from COVID-19 patients. Then, human related physiological models are urgently needed to investigate these questions and for functional validation. In agreement with these arguments, an early study on SARS-CoV indicates that host cell differentiation or polarized epithelium and expression of ACE2 are both important for the susceptibility of human airway epithelia to SARS-CoV viral infection (Jia et al. 2005; Tseng et al. 2005). Indeed, ALI cultures of human airway epithelial cells (HAEs) have also been used for functional drug screening of SARS-CoV and SARS-CoV-2 (Sheahan et al. 2017; Sheahan et al. 2020). Since CRCs are stable resources for normal functional airway cells, CRCs/ALI cultures will facilitate these studies and development of novel therapeutics as a functional and biological system.', 'As we discussed above, CR is a rapid, robust and sufficient way to obtain large amount of normal airway cells from non-invasive and minimal human samples, and CRCs from respiratory tract maintain their lineage functions. ALI provides a unique environment to mimic “in vivo” physiological conditions. Thus, CRC/ALI will be appropriate for studies of viral entry to the nature host cells, innate immune response of the host cells, and tests of anti-viral and immune (innate) modulators (Fig. 3). This will be a unique biological or functional cell system for population or health disparity studies since there is no biological cell model available in the field. Since CRCs can be easily genetically manipulated with overexpression of exogenous genes, shRNAs, or CRISPR technologies (Chu et al. 2015a; Fenini et al. 2018), CRC/ALI should be a unique system for both phenotypical and mechanism studies for human viral diseases. Although CRCs alone can be used for rapid and high throughput screening, combination CRC and ALI will take much longer time and need further development for high throughput studies. Compare to the current permissive cancer cell lines or animal derived permissive cell lines, CRC/ALI for virus studies requires experienced performance, higher work load and cost. Thus, CRC/ALI will serve as a human-relevant, physiological system for emerging viruses in addition the current cell lines and animal models.', 'Because of limitations of current cell line and animal models, there is an urgent need of human physiological cell models for studies of emerging viruses. Here we summarized establishment of long term cultures for human normal airway epithelial cells from nose to lung using CR and coupled ALI technologies and their applications as an ex vivo model for emerging viruses. As a great addition to the current cell lines and animal models, CRC/ALI system will facilitate studies of viral entry including receptors and internalization, innate immune responses, viral replications, and drug discovery.\\n']\n",
            "['Individuals with spinal cord injuries and disorders (SCI/D) have an elevated risk of death with community-acquired pneumonia [1], and there may be elevated risk with coronavirus disease 2019 (COVID-19). The Veterans Health Administration (VHA) SCI/D National Program Office oversees health care services received by over 24,000 Veterans with SCI/D at 170 medical centers and more than 1000 outpatient care sites. In coordination with the VHA Support Service Center, the National Program Office maintains the VHA SCI/D Registry, which identifies Veterans who have received health care services from the VHA SCI/D System of Care. To support outreach and prevention activities in the response to the current COVID-19 pandemic, the National Program Office developed operational reports showing COVID-19 infection status as determined by occurrence of a positive laboratory test performed by VHA or by entry of a non-VHA lab result. Additional registry data including demographics, neurologic classifications, comorbidities, and geographic setting were used to determine a preliminary infection risk category and severity risk category for each SCI/D Veteran. These reports were developed in March 2020 and made available to VHA SCI/D Centers on April 7, 2020. The objective of this analysis was to describe case fatality of COVID-19 infection in Veterans with SCI/D as determined using these operational reports.', 'Between March 9, 2020 and June 30, 2020, COVID-19 was identified in 140 of 17,452 Veterans with SCI/D, inclusive of traumatic and nontraumatic etiologies and excluding both multiple sclerosis and amyotrophic lateral sclerosis. The mean (SD) age of those with infection was 67 (12) years. The racial composition was White: 80 (57%), Black: 50 (36%), and other/declined/unknown 10 (7%). For those with neurologic classifications recorded (n = 82), the combined neurological level and American Spinal Injury Association Impairment Scale (AIS) categories were: C1–C4 tetraplegia AIS A–C: 13 (16)%; C5–C8 tetraplegia AIS A–C: 14 (17%); paraplegia AIS A–C: 15 (18%); all AIS D: 40 (49%). Infections were identified in Veterans from 63 VHA health care systems. Inpatient care was received by 67 of 140 (48%) Veterans in VHA hospitals and by an unknown number of Veterans in other inpatient settings.', 'Of these 140 Veterans who tested positive for COVID-19, 26 (19%; 95% CI, 13–26%) had died as of June 30, 2020, with all except one death occurring within 30 days of testing positive. For the 19 deceased Veterans who received VHA inpatient care, the median duration between positive test and death was 8 days. The case fatality rate of 19% in Veterans with SCI/D can be compared with the rate in the overall US Veteran population enrolled for VHA health care, which has a similar proportion of individuals age 65 years or greater (SCI/D: 54%; overall population: 49%) [2]. As of June 30, 2020, 7.7% of all non-SCI/D Veterans known to have COVID-19 had died: 1564 deaths in 20,369 infected individuals, calculated as overall Veteran population minus SCI/D Veteran subtotals [3]. The SCI/D Veteran case fatality rate with COVID-19 is 2.4 times the rate observed in the non-SCI/D Veteran population, with an absolute rate that is 11% greater (95% CI: 5–19%; Z score = 4.8; p < 0.0002).', 'These case fatality rates for Veterans with SCI/D and the overall Veteran population, as well as what has been reported elsewhere for the US population and the world, are overestimates of the infection fatality rate. In each of these populations, individuals with asymptomatic infections are unaccounted for, and individuals with minimal symptoms may not have received testing, particularly early in the pandemic when testing capacity in the US was minimal. Also, it is possible that mild symptoms from COVID-19 in an individual with SCI/D could be misattributed to other medical conditions commonly experienced, such as urinary tract infection. This would bias diagnosis of COVID-19 in individuals with SCI toward cases with greater severity and risk of fatality.', 'Despite these limitations, these data from a large registry population provide preliminary evidence that individuals with SCI/D have higher case fatality and infection severity with COVID-19 than the general population. The VHA SCI/D population includes a considerable number of older individuals with multiple medical comorbidities and thus case fatality is likely substantial. These findings support current efforts to minimize the risk of outbreaks of COVID-19 in institutional settings caring for individuals with SCI/D. Individuals with SCI/D in all settings should be strongly encouraged to take actions to minimize their chance of acquiring COVID-19.']\n",
            "['Colorectal cancer (CRC) is the fourth most common tumor and the fourth most common cause of tumor-associated mortality worldwide, accounting for an estimated 1.4 million new cases and 700,000 deaths in 2012.[1,2] The continued development of modalities for the early diagnosis and prevention of CRC has tremendously benefited patient care. However, there were an estimated > 1.8 million new cases of CRC and 881,000 deaths in 2018, which accounted for about 10% of all newly diagnosed cancers and related deaths.[3] This tendency indicates the urgency for the early diagnosis and treatment of CRC. In addition, the 5-year relative survival rate of CRC patients remains at less than 50% in underdeveloped countries, whereas the survival rate is approximately 65% in developed countries, including Canada, the United States, Australia, and several European countries.[4–6] Surgical resection plays a significant role in the management of colon cancer.[7] For stage III colon adenocarcinoma (COAD), adjuvant systemic treatment is crucial to decrease the incidence of recurrence and increase the overall survival (OS) rate.[8,9] For patients with node-negative metastatic colon cancer, administration of adjuvant systemic treatment can greatly decrease tumor recurrence and increase patient survival.[10] However, the initiation of adjuvant systemic treatment is unnecessarily delayed in some cases.[11]', 'Long noncoding RNAs (lncRNAs), mRNA-like molecules that lack open reading frames, are highly evolutionarily conserved noncoding RNAs with lengths of > 200 nucleotides that have emerged as integral components of mammalian transcription.[12,13] Current studies have found that lncRNAs play crucial roles in the transcription of structural and functional proteins, thus the potential roles of lncRNAs have become an important focus in medical science.[13] Research suggests that lncRNAs play pivotal roles in a variety of cellular biological processes (BPs), especially in the regulation of cellular apoptosis,[14] proliferation, differentiation, and migration,[15] as well as genomic imprinting,[16] gene expression, posttranslational modification, and tumorigenesis.[17,18]', 'Many recent studies have investigated the clinical significance of lncRNAs in tumor progression. For instance, lncRNA SNHG7 is reportedly involved in varied pathological processes in different tumor types, such as hepatocellular carcinoma,[19] renal cell carcinoma,[20] and lung cancer,[21] in addition to CRC and COAD. Furthermore, lncRNA MALAT1 was found to promote cellular invasion and metastasis in CRC by sponging with microRNA-106b-5p[22] and lncRNA LINC01234 was reported to promote the expression of serine hydroxymethyltransferase 2 and cellular proliferation in colon cancer.[23] However, the roles of lncRNA FRGCA in tumorigenesis remain relatively unknown. Therefore, the aim of the present study was to explore the potential utility of lncRNA FRGCA as a diagnostic and prognostic indicator of COAD.', 'The cohort of this retrospective study was limited to patients with pathologically confirmed COAD. Data regarding the expression profiles of lncRNA FRGCA were download from The Cancer Genome Atlas database (https://cancergenome.nih.gov/) and were normalized using the DESeq module of the R/Bioconductor package.[24] Clinical data coupled to the expression data of mRNAs, miRNAs, and lncRNAs related to COAD were downloaded from the Oncolnc database (http://www.oncolnc.org/).', 'To determine the diagnostic significance of lncRNA FRGCA in COAD, the expression profiles of tumor and nontumor tissues were determined, and a receiver operating characteristic (ROC) curve was constructed. An area under curve (AUC) of ≥0.7 was considered to demonstrate diagnostic significance. On the basis of the expression patterns of lncRNA FRGCA in tumor tissues, the COAD patients were divided into 2 groups: low vs high lncRNA FRGCA expression. Kaplan–Meier plots were constructed to reveal the prognostic significance of lncRNA FRGCA in COAD. Adjusted probability (P) values were calculated to determine whether lncRNA FRGCA was significant for the prognosis of COAD. Clinical factors with P values < .5 were considered significant and thus used for adjusted survival analysis.', 'Cases were divided into 4 groups according to tumor stage (I–IV) and t tests were performed to identify differences in lncRNA FRGCA expression levels between stage I vs II, III, and IV. Here, significant P values demonstrated that lncRNA FRGCA expression was associated with COAD progression. A forest plot was constructed using GraphPad software, version 7.0 (GraphPad Software, Inc., La Jolla, CA), to visualize hazard ratios (HRs) and P values of each factor associated with COAD progression.', 'Stratified analysis of clinical factors was performed to determine the prognostic significance of lncRNA FRGCA in COAD. The following factors were included for analysis: sex (male vs female); age (≤60 vs >60 years); and tumor stage (I vs II, III, and IV).', 'Nomograms were constructed based on age, sex, tumor stage, and lncRNA FRGCA expression to predict the 1-, 3-, and 5-year OS rates of COAD patients. Each factor was assigned a score, which were summed for survival prediction.', 'The Pearson correlation coefficient (r) was evaluated to identify correlations between lncRNA FRGCA and mRNAs throughout the genome using R version 3.5.0 (https://www.r-project.org/). The mRNAs correlated with lncRNA FRGCA (r ≥ .25) were used for further analysis. Enrichment analysis of these mRNAs was performed to identify Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways and gene ontology, especially those related to the functions of BPs, cellular components (CCs), and molecular functions (MFs), using the online Database for Annotation, Visualization and Integrated Discovery (DAVID, https://david.ncifcrf.gov/home.jsp) version 6.8.[25,26] A GGI network was constructed to visualize the interactions among the mRNAs of interest using the GeneMANIA plugin of the Cytoscape software platform,[27,28] while visualized BP, CC, and MF analyses were performed using the BinGO plugin.[29]', 'Statistical analysis was performed using SPSS Statistics for Windows, version 17.0 (SPSS, Inc., Chicago, IL). The Kaplan–Meier method was used to determine the median survival time, while the log-rank test was used to provide P values and identify differences among groups. Both univariate and multivariate Cox proportional hazards regression models were constructed, and 95% confidence intervals (95% CIs) and HRs were calculated. A P value ≤.05 was considered statistically significant.', 'The study cohort included a total of 438 COAD patients (234 males and 204 females), of whom 134 were aged ≤60 years and 302 were aged >60 years (age data were missing for two patients). In regard to tumor stage: 73 were classified as stage I, 167 as stage II, 126 as stage III, and 61 as stage IV (tumor stage data was missing for 3 patients). Among these factors, only tumor stage was significantly correlated to OS (Table 1). Thus, the multivariate Cox proportional hazards regression model was further adjusted for tumor stage.', 'As compared to nontumor tissues, lncRNA FRGCA was highly expressed in tumor tissues (Fig. 1A, P < .001) and was differentially expressed in the low vs high group (Fig. 1B, P < .001).', 'In the diagnostic significance analysis, a ROC curve showed that the AUC of lncRNA FRGCA was 0.763 with a sensitivity (Se) of 97.56% and specificity (Sp) of 52.08%, demonstrating diagnostic significance of lncRNA FRGCA in COAD (Fig. 2A, P < .0001, 95% CI = 0.711–0.815). Although the univariate Cox proportional hazards regression model showed that lncRNA FRGCA had no prognostic value in COAD (crude P = .086), the multivariate model did reveal prognostic value (adjusted P < .001, HR = 0.444, 95% CI = 0.288–0.685, Table 2, Fig. 2B). These results showed that high expression of lncRNA FRGCA indicated better survival in COAD.', 'Stratified analysis was performed to further explore the prognostic significance of these factors in relation to lncRNA FRGCA in COAD. Kaplan–Meier plots of sex (male vs female), age (≤60 vs >60 years), and tumor stage (I vs II, III, and IV) are presented in Figure 3 and Table 3. The factors male, female, age ≤ 60 and >60 years, and tumor stages II and IV were associated with the OS of COAD patients (all, adjusted P < .05). In addition, a forest plot was constructed to visualize the HR and P values of these factors (Fig. 4).', 'Nomograms were constructed using the factors age, sex, tumor stage, and lncRNA FRGCA expression. As shown in Figure 5, the factors female, age > 60 years, tumor stages III and IV, and low expression of lncRNA FRGCA had higher scores than male, age ≤60 years, tumor stages I and II, and high expression of lncRNA FRGCA. Differences in total scores indicated variations in the prediction of 1-, 3-, and 5-year OS, where a higher score was correlated to poorer prognosis at all time points. Moreover, these scores indicated that OS was best at 1 year and poorest at 5 years.', 'Pearson correlation was calculated between lncRNA FRGCA and PCGs. Detailed mRNA names and coefficients of correlation are listed in Supplementary Table 1. A GGI network of coexpression relationships among these mRNAs is shown in Figure 2D.', 'Enrichment analysis using DAVID revealed that the BP, CC, and MF functions were enriched in regard to the mitochondrial inner membrane, mitochondrion, mitochondrial translational elongation, mitochondrial translational termination, structural constituents of the ribosome, protein binding, translation, ribosome, and nucleoplasm, among others (Fig. 6A, B, D). The following KEGG pathways were enriched: ribosome, Huntington disease, metabolic pathways, pyrimidine metabolism, oxidative phosphorylation, Alzheimer disease, Parkinson disease, RNA polymerase, nonalcoholic fatty liver disease, and the sulfur relay system, among others (Fig. 6C). Detailed enrichment results of BP, CC, and MF, and KEGG pathways are shown in Supplementary Tables 2 and 3, respectively.', 'Furthermore, the visualized BP, CC, and MF results are shown in Figures 7–9, respectively. The BP-related functions included cellular metabolic processes, primary metabolic processes, CC organization, oxidative phosphorylation, ribonucleoprotein complex biogenesis, protein complex assembly, and RNA modification, among others (Fig. 7). The CC-related functions included intracellular membrane-bound organelles, cytoplasm, large ribosomal subunit, mitochondrion, intracellular part, intracellular organelle, mitochondrial envelope, and cytoplasmic part and organelle lumen, among others (Fig. 8). The MF-related functions included structural constituents of the ribosome, structural molecule activity, nuclease activity, RNA polymerase activity, hydrolase activity, hydrolyzing N-glycosyl compounds, NADH dehydrogenase activity, oxidoreductase activity, acting on NADH or NADPH, and quinone or similar compounds as the acceptor, among others (Fig. 9).', 'The results of the present study revealed that lncRNA FRGCA was differentially expressed in COAD tumor and nontumor tissues. Further analysis demonstrated that lncRNA FRGCA had diagnostic and prognostic significance in COAD. High expression of lncRNA FRGCA indicated better survival in COAD patients. In addition, stratified analysis found that sex, age, and tumor stage were associated with lncRNA FRGCA expression. Moreover, a nomogram was constructed using sex, age, tumor stage, and lncRNA FRGCA expression to predict 1-, 3-, and 5-year OS rates. However, tumor stages II, III, and IV were not associated with lncRNA FRGCA expression.', 'Pearson correlation was employed to identify correlations with mRNAs throughout the genome, which were than selected for enrichment analysis. Enrichment analysis of BP, CC, and MF indicated that these mRNAs were involved in mitochondrial inner membrane and translational termination, structural constituents of the ribosome, protein binding, translation, ribosome, and nucleoplasm, among others. Enrichment analysis of KEGG pathways indicated that these mRNAs were involved in ribosome, oxidative phosphorylation, metabolic pathways, pyrimidine metabolism, RNA polymerase, nonalcoholic fatty liver disease, and the sulfur relay system, among others. Therefore, we hypothesized that lncRNA FRGCA was associated with the diagnosis and prognosis of COAD via oxidative phosphorylation, ribosome metabolism, and metabolic pathways, especially in the nucleoplasm.', 'In 2002, Okazaki et al[30] first reported the involvement of lncRNAs in mammalian transcription. Since then, the importance of these nonprotein coding molecules has been widely investigated.[30] For example, in the Encyclopedia of DNA Element study, Djebali et al[31] revealed that the human genome has more than 9640 lncRNA loci, which accounted for approximately one-half of all protein-coding genes. The findings of prior studies have gradually altered our attitude and understanding of the mammalian genome and raised new questions of the potential functions of lncRNAs.[13] The biogenesis of lncRNAs parallels that of mRNAs, as a vast proportion are transcribed by RNA polymerase II and have a 5’ cap and 3’ polyadenylation signal.[13] Combined with prior analysis of transcriptional outputs, sequencing of the human and mouse genomes indicated that approximately 80% of the transcription of the mammalian genome occurs in a cell-specific manner, which gave rise to a new understanding of transcriptional modulation, especially within noncoding regions.[31,32]', 'The use of integrated methodologies with the continued progress in RNA sequencing methods has facilitated the discovery and identification of many new lncRNAs. However, other than carcinoembryonic antigen and carbohydrate antigen 19–9, there is currently a lack of biomarkers for the early diagnosis of CRC, primarily because of the relatively low Sp and Se.[33–36] A previous molecular study of stool samples has identified several diagnostic biomarkers with relatively high Se and Sp for colon cancer.[37] The many advantages of scatological studies include no requisite of bowel preparation, testing on an outpatient basis, the noninvasive nature of the assays, ease of performance, and no risk of complications that commonly occur with colonoscopy, especially perforation and bleeding. Hence, the use of fecal samples for colon cancer screening offers a noninvasive and relatively inexpensive option to the patient.[37] More surprisingly, these fecal immunochemical tests have greater Se and Sp than colonoscopy. In fact, Weller et al[38] reported a Se of 83% and Sp of 96% for screening of CRC among 6208 subjects; Rozen et al[39] reported a Se of 86% and Sp of 98% for screening of CRC and advanced adenoma (AA) in 403 subjects; Wong et al[40] reported a Se of 62% and Sp of 93% for screening of CRC and AA among 250 subjects at a cutoff of 70 standard units; Lohsiriwat et al[41] reported a Se of 91% and Sp of 94% for screening of CRC among 164 subjects; and Rozen et al[42] reported a Se of 69% and Sp of 92% for screening of AA or CRC among 330 subjects at a cutoff value of 50 ng/dL. Meanwhile, the results of the present study showed that lncRNA FRGCA had diagnostic significance with AUC of 0.763, with a Se of 97.56% and Sp of 52.08% among 438 COAD patients. Although there were differences from obvious COAD tissues, our results lead to a novel finding of lncRNA FRGCA as a potential diagnostic biomarker for COAD. Of course, the diagnostic significance of lncRNA FRGCA must be further validated in multicenter studies.', 'Colon cancer-associated transcript-1 (CCAT1), an lncRNA containing 2628 bp, is located on chromosome 8q24.21 and has been reported to facilitate the progression of colon cancer, as it was abnormally expressed in tumor tissues.[43] Another investigation revealed that CCAT1 transcription was stimulated by c-Myc, and then facilitated the growth and invasion capacities of colon cancer cells.[44] CCAT1 expression was notably increased in the early and late stage of colon cancer with significant upregulated expression in adenomatous polyps, the tumor-proximal colonic epithelium, sites of liver metastasis, and the lymph nodes.[45] In addition to prognosis, CCAT1 expression is reportedly associated with tumor stage, local infiltration depth, vascular invasion, and carbohydrate antigen 19–9 levels.[46]', 'Furthermore, the functional characteristics of CCAT1 in other tumors have also been explored. Previous studies have reported that CCAT1 expression is consistently upregulated and functions as an oncogene in cancers of the stomach,[47,48] liver,[49,50] gallbladder,[51] ovary,[52] breast,[53] and lung.[54,55] Moreover, CCAT1 was overexpressed in hepatocellular carcinoma tissues and promoted cell invasion, proliferation, and migration in vitro.[56] CCAT1 has been shown to promote cell proliferation in lung cancer.[54] Once activated by c-Myc, CCAT1 promotes the progression of gastric cancer,[47] while overexpression of CCAT1 in breast cancer tissues has been associated with tumor-node-metastasis staging, lymph node metastasis, and tumor differentiation, as indicated by poor OS and tumor progression-free survival.[45] To the best of our knowledge, this is the first reported association between lncRNA FRGCA expression and tumor progression. Our results confirmed the prognostic significance of lncRNA FRGCA in COAD, as high expression was correlated with better OS. However, in this study, there was no association between lncRNA FRGCA expression and tumor stage. As with other lncRNAs, CCAT1 has been associated with various other tumors, thus we speculate that lncRNA FRGCA expression could be used as a prognostic indicator with other tumors as well, although further studies are needed to explore such possibilities.', 'Abnormal regulation of lncRNAs has been shown to contribute to tumor pathologies, which also provide a new clue for the lncRNA-based treatment.[57–59] The results of the present study revealed that lncRNA FRGCA-related mRNAs were enriched in the processes of mitochondrial inner membrane and translational termination, structural constituents of the ribosome, protein binding, translation, neoplasm, oxidative phosphorylation, metabolic pathways, and RNA polymerase, among others. Therefore, we speculate that lncRNA FRGCA expression is a suitable indicator for the diagnosis and prognosis of COAD, based on the relationships with oxidative phosphorylation, ribosome metabolism, metabolic pathways, especially in the nucleoplasm, although further functional trials are needed for validation of these findings.', 'There were some limitations to this study that should be addressed. First, the study cohort was relatively limited, thus a larger population is needed to validate the diagnostic and prognostic significance of lncRNA FRCGA in COAD. Second, the clinical value of lncRNA FRCGA on COAD must be investigated in multicenter studies across countries and regions. Third, further functional trials are needed to explore the specific mechanisms underlying the involvement of lncRNA FRCGA in COAD and to investigate potential associations between lncRNA FRGCA and the progression of other tumors. In addition, consensus molecular subtypes should be performed for further understanding the mechanism of COAD in future studies.', 'The results of the present study reported, for the first time, associations between lncRNA FRGCA expression and COAD prognosis and diagnosis. In addition, potential mechanisms were identified. We found that lncRNA FRGCA expression was greater in tumor tissues than nontumor tissues, suggesting diagnostic and prognostic value. Furthermore, a nomogram was created to predict the 1-, 3-, and 5-year OS rates. Functional enrichment analysis of genome-wide lncRNA FRGCA-related mRNAs revealed involvement in mitochondrial translational termination, ribosome, protein binding, translation, oxidative phosphorylation, metabolic pathways, among others. Therefore, we speculate that the associations between lncRNA FRGCA and COAD occur through ribosome metabolism, oxidative phosphorylation, and nucleoplasm-related metabolic pathways. Nonetheless, further studies are needed to validate these results and our hypothesis of the association between lncRNA FRGCA and tumor progression in COAD.', 'Cun Liao and Feng Gao designed this manuscript; Yun Guo, Yizhen Gong, Xue Huang, Xiwen Liao, Xiangkun Wang, and Guotian Ruan conducted the study and analyzed the data. Cun Liao wrote the manuscript and Feng Gao guided the writing and revision.']\n",
            "['Human rhinoviruses (HRVs) are highly prevalent respiratory pathogens traditionally associated with mild and self-limited upper respiratory tract infections (‘colds’). The replacement of relatively insensitive virus isolation methods with molecular methods for HRV detection has contributed to a recent reappraisal of their incidence of infection and genetic diversity. HRVs are now known to be intimately linked to the development and exacerbations of respiratory diseases such as asthma (Khetsuriani et al., 2008; Miller et al., 2007) and responsible for more severe disease manifestations such as bronchiolitis in young children and in the immunosuppressed (Gutman et al., 2007; Ison et al., 2003; McErlean et al., 2007). Genetic characterization of HRVs detected by molecular methods has revealed much greater diversity than previously described; approximately one-third of HRV infections are now known to be caused by a third group of rhinoviruses (species C), uncultureable in vitro and therefore entirely missed by traditional virus isolation methods (Arden et al., 2006; Kaiser et al., 2006; Kistler et al., 2007a; Lamson et al., 2006; Lau et al., 2007; Lee et al., 2007; McErlean et al., 2007; Olenec et al., 2010; Renwick et al., 2007).', 'HRVs are members of the genus Enterovirus of the family Picornaviridae, a large virus group that additionally contains human enteroviruses and a number of viruses infecting non-human primates, pigs, cattle, sheep and other mammals (Knowles et al., 2012). The genus Enterovirus has recently been divided into a total of 12 species based on their genetic divergence, sites of replication and disease associations (http://www.picornaviridae.com). Among these species, three are designated Rhinovirus A, Rhinovirus B and Rhinovirus C, a nomenclature that reflects their primary tropism for the respiratory tract. Currently, humans are the only known natural host, although it is possible that primates could also be susceptible. Rhinovirus B and most Rhinovirus A variants use the intracellular adhesion molecule-1 (major) receptor for cell entry (Uncapher et al., 1991), whilst a subset of Rhinovirus A types use the low-density lipoprotein receptor (Hofer et al., 1994). The receptor(s) used by HRV species C (HRV-C) types is currently unknown. HRVs are most commonly transmitted by the respiratory–salivary route, both by person-to-person contact and by airborne transmission. In temperate countries, infections occur primarily in two peaks, the first between April and May and the second between September and October (Monto, 2002; Vesa et al., 2001).', 'HRVs within each species show remarkable genetic and, where determined, antigenic heterogeneity. Species A and B isolates were originally characterized serologically, with 75 and 25 serotypes being classified by cross-neutralization assays by 1987 (Hamparian et al., 1987; Kapikian et al., 1967, 1971), and prototype strains of each were lodged at the American Type Culture Collection (ATCC). However, as well as being laborious, time-consuming and requiring extensive panels of antisera specific to each serotype, classification by neutralization cannot be used for species C rhinoviruses because of their refractoriness to cell culture.', 'Human enteroviruses (EVs) are classified into four species (Enterovirus A, Enterovirus B, Enterovirus C and Enterovirus D) and, like HRVs, isolates have also been traditionally characterized by neutralization assays. Some neutralization determinants are located on the external VP1 protein in the virus nucleocapsid, and it has been shown that sequence divergence in this region correlates with serotype designation of poliovirus isolates (Kilpatrick et al., 1998) and other enteroviruses (Kiang et al., 2009; Oberste et al., 1999a, b). Indeed, a threshold of 25 % nucleotide and 12 % amino acid sequence divergence generally corresponds to the previous serological division of EVs into members of the same or different serotypes (Oberste et al., 1999a). As well as providing a simpler method for EV type identification than the previously used serological assays, VP1 sequencing is now routinely used for identification and assignment of new EV types without serological confirmation of antigenic differences (Brown et al., 2009; Oberste et al., 2002, 2004, 2007).', 'In the current study, we investigated whether a comparable threshold of sequence diversity in the VP1 of HRVs would enable a similar genotypic division of HRV-A, -B and -C variants into types and additionally whether this correlated with sequence divergence in the VP4/VP2 region (routinely used for HRV typing). This investigation was prompted by the increasing use of molecular methods to characterize HRVs and the consequent lack of isolates or antisera with which to antigenically characterize them and, secondly, by the difficulty in culturing species C HRVs required for neutralization assays (Bochkov et al., 2011; Kistler et al., 2007a; McErlean et al., 2007).', 'This investigation follows on from previously published proposals for the classification of HRV-C based on a 13 % divergence threshold in VP1 (Simmonds et al., 2010). These have been adopted by the Picornaviridae Study Group as the primary mechanism for defining HRV-C types. We have extended this classification system for species A and B rhinoviruses. The previously used term ‘serotype’ reflects the identification and classification of HRV variants by antigenic properties and therefore assignment by genetic comparison methods requires a different or combined term. Following the terminology now used for enteroviruses, we have adopted the term ‘type’ throughout the study to represent those HRV variants that have been identified and classified by either cross-neutralization or genetic comparisons.', 'In order to determine whether a distinct threshold that divided pairwise p-distance comparisons in the VP1 region into intra- and inter-type values existed within each HRV species, distributions of pairwise nucleotide p-distances were constructed from a total of 435 HRV-A, 133 HRV-B and 206 HRV-C VP1 sequences (Fig. 1; sequences used for species C represent an expanded dataset from the one analysed previously; Simmonds et al., 2010). These showed a maximum within-species nucleotide p-distance of 39.4, 33.9 and 42.9 % for species A, B and C, respectively, and minimum values of between 10 and 14 % divergence. Detailed inspection allowed identification of minimum values in the distribution and assignment of a nucleotide divergence threshold for each species (Fig. 1, right-hand panels). A threshold of 13 % was identified for both HRV-A and -C sequences, whilst the less-divergent HRV-B isolates showed a minimum value that supported a 12 % threshold.', 'The total datasets of available VP1 sequences contained unequal representations of different types within each species, ranging from one to a maximum of 13 for HRV-A (A49), 10 for HRV-B (B69) and 22 for HRV-C (C3). Particularly for pairwise distance comparisons, these may potentially distort the shape of the distribution of inter-type distances because comparisons with certain types are over-represented in the distance totals. An additional analysis was therefore carried out with only single sequences from each type (grey shading superimposed in Fig. 1, left graphs). For each, prototype strains of HRV-A and -B were those of the first full-genome sequence of a deposited ATCC isolate that was deposited in GenBank (highlighted in red in Table S1 in JGV Online). Prototype strains of HRV-C were those listed previously (Simmonds et al., 2010; Table S1). This closely reproduced the analysis of the whole VP1 dataset in the shape and range of the inter-type distribution of distances (Fig. 1). Whilst distance distributions between VP1 sequences of HRV-A and -C showed a single peak, those of HRV-B showed a bimodal distribution. The peak between p-distance values of 0.19 and 0.26 corresponded to comparisons between HRV-B types within each of the four basally branching phylogenetic clusters (indicated by grey-outlined boxes in Fig. 2). The larger peak thus corresponded to comparisons between variants in different clusters.', 'Phylogenetic trees were constructed of the VP1 region of HRV-A, -B and -C (Fig. 2). All HRV sequences fell into bootstrap-supported monophyletic groups that closely matched types assigned by sequence distances. These groups were numbered either by clustering with a known prototype strain (HRV-A/B 1–100), by order of submission of full-genome (HRV-A101–106, HRV-B100–104 and HRV-C1–11) or VP1 sequences (HRV-C12–51). The vast majority of sequences within each terminal clade showed intra-clade VP1 divergence substantially less than the proposed thresholds of 13 % in HRV-A and -C and 12 % in HRV-B (Table 1). Most of these also showed a minimum inter-clade VP1 divergence (with a different assigned type) that was much greater than the proposed threshold.', 'Consistent with the thresholds identified for each species, three new types of HRV-A and three of HRV-B have been proposed previously; HRV-A101 (Rathe et al., 2010), HRV-A102 (de Vries et al., 2008), HRV-A103, HRV-B100 (Linsuwanon et al., 2011), HRV-B101 and HRV-B102 (McIntyre et al., 2013) (http://www.picornaviridae.com). In addition, the current analysis provided evidence for three further new HRV-A types (tentatively labelled HRV-A104–A106) and two novel HRV-B types (HRV-B103 and -B104), labelled in red in Fig. 2. The additional sequence data provided in this study and published elsewhere for HRV-C now create a phylogenetic tree that can be divided into 51 distinct clades. These have been assigned as types based on sequence divergence in the VP1 region (Table 1), adding a further 18 types to the total of 33 first defined in 2010 (Simmonds et al., 2010) (http://www.picornaviridae.com).', 'Although several of the putative new HRV-A and -B types included showed minimum VP1 distances far greater than the assignment thresholds shown in Fig. 1, several were close to or below the 13 and 12 % levels (Table 1). As examples of the former, proposed new HRV-B types B100, B101 and B102 showed minimum pairwise p-distances in VP1 of 0.214, 0.190 and 0.227 from their nearest confirmed HRV-B type (HRV-B35). Pairwise nucleotide p-distances between these three new HRV-B types ranged from 0.163 to 0.171. All proposed species C types were similarly clearly above the assignment threshold of 13 % for this species (Table 1). However, other variants proposed as new types showed minimum distances that were much closer to the threshold; for example, the six available sequences from the novel group putatively designated HRV-B103 showed VP1 distances from their nearest neighbour type (HRV-B6) ranging from 0.113 to 0.148. However, for these two types, only two pairwise comparisons fell below the 12 % threshold and both involved sequence F110_9318.', 'The newly designated HRV-B104 represented a division of a previously single type, HRV-B52, based on the existence of two discrete phylogenetic clades with VP1 p-distances of 0.121–0.142 between them. In this case, the clade containing the original ATCC isolates, including strains FJ445188 (GenBank accession no.) and EF173424, has retained the HRV-B52 designation, whereas the clade containing field isolates FJ445137 and JF781506 has been reassigned as HRV-B104. Similar variability was observed in species A where two of the proposed new types showed distances close to the assignment threshold (HRV-A104 and -A105).', 'Analysis of pairwise VP1 divergence between previously recognized HRV-A types also revealed several examples of VP1 distances below the 13 % threshold. For example, HRV-A8 and HRV-A95 showed an inter-type VP1 p-distance range of 0.015–0.019. We propose that all sequences of these two types are reassigned as HRV-A8, a proposal consistent with their previously documented serological cross-reactivity (Ledford et al., 2004) (Fig. 3a).', 'Similarly, HRV-A29 and HRV-A44 have previously demonstrated cross-reactivity (Cooney et al., 1982) and prototype strains showed an inter-type VP1 p-distance of 0.0724 (Fig. 3a). We propose that these sequences be reassigned as HRV-A29. Interestingly, contemporary sequences that belonged to HRV-A29 by VP1 p-distance from the prototype strains (maximum p-distance of 0.127) showed a maximum divergence of 0.159 when compared with each other. All contemporary HRV-A29 sequences grouped separately on phylogenetic trees from both the HRV-A29 and HRV-A44 prototype sequences (Fig. 3a).', 'HRV-A54 and HRV-A98 are a third type pair that are more closely related than expected, with a pairwise nucleotide VP1 p-distance of 0.107–0.124 (Fig. 3a). Although cross-neutralization properties of these two types have not been analysed in previous studies (Cooney et al., 1982; Ledford et al., 2004), we propose that all variants classified as these two types are reassigned as HRV-A54.', 'In addition to those type pairs that have been combined into single type assignments, a number of other HRV types displayed intermediate divergence values that violated the proposed thresholds. These included HRV-A1A and -A1B, which are serologically cross-reactive (Cooney et al., 1982) but showed several pairwise p-distances that fell above the VP1 threshold (Fig. 3). We propose that members of these two clades retain their assignment as HRV-A1, although this should be reviewed as more sequences and confirmatory serological data become available. As a contrasting problem, the type pair HRV-A25 and -A62 showed VP1 pairwise distances ranging from 0.093 to 0.103 from each other (Fig. 3b), below the assignment threshold, despite showing no cross-neutralization (Cooney et al., 1982). We propose leaving these assignments unchanged pending confirmation of the serological relationship between these two types. Similarly, prototype isolates of types HRV-A36 and -A89 (GenBank accession nos DQ473508 and M162488) showed VP1 p-distances of 0.1292, just below the assignment threshold (Fig. 3b). For these types, the HRV-A36 antiserum can neutralize HRV-A89 (as well as HRV-A50 and HRV-A58). However, the antiserum to HRV-A89 cannot neutralize HRV-A36 and almost all pairwise distances between more recent isolates of HRV-A36 (n = 9) and HRV-A89 (n = 5) are above the assignment threshold. We propose to retain their current designations as HRV-A36 and HRV-A89.', 'Only one HRV-C type showed p-distance values that violated proposed thresholds. HRV-C8 strains were divided into two clades, one previously classified as HRV-Cpat28 and both containing contemporary sequences (Fig. 3b). Although the VP1 p-distance between the two clades ranged from 0.129 to 0.132, we propose that these two clades should remain classified as HRV-C8 on the basis of borderline VP1 divergence.', 'Despite these examples, the number of pairwise distances in the overall distribution that were close to the assignment threshold (and therefore potentially ambiguously assigned) was small (Fig. 4). The majority represented comparisons between type pairs that should be combined (Fig. 4, green). Only six HRV-A types showed any between-type distances that were below the threshold (pink), and there were three with within-type distances that were above (Fig. 4, purple). As described above, their current classification can incorporate phylogenetic data and any available information on neutralization properties in addition to VP1 distances. Combined, these may potentially resolve any remaining uncertainties over type assignments.', 'At the conclusion of this analysis of all available HRV VP1 sequences, totals of 77 HRV-A types, 29 HRV-B and 51 HRV-C types have been listed in full in Tables S1 and S2). Novel strains of HRV were classified based on date of submission to the Picornaviridae Study Group, rather than the earliest isolated strain.', 'The well-conserved and readily amplified VP4/VP2 region of HRV is commonly used in studies of its epidemiology and clinical associations, with over 3900 HRV sequences from this region in GenBank, over 10 times the number of VP1 sequences (n = 384). To investigate the value of VP4/VP2 sequences for HRV type identification, we performed phylogenetic analysis on the entire dataset and compared groupings of sequences of known type to those observed in VP1 (Fig. 5).', 'The majority of HRV strains grouped congruently between VP1 and VP4/VP2. Only one sequence showed evidence of phylogenetic incongruence; JX291115 is a member of HRV-C51 type in the VP1 region, with a pairwise nucleotide p-distance of 0.01 to JF317015 (HRV-C51 prototype strain). However, in VP4/VP2, this sequence showed a VP1 p-distance of 0.146 from JF317015 and lay closest to HRV-C15. Its status as a recombinant sequence was demonstrated by triplet analysis using the rdp software package. In contrast, comparable analysis of all other HRV-C and HRV-A/B variants with both full VP4/2 and VP1 sequences available by rdp and gard (164 types in species A, 70 in species B and 145 in species C) demonstrated no evidence of recombination between regions.', 'We previously proposed a divergence threshold of 10 % for identifying different HRV-C types using VP4/VP2 sequences (Simmonds et al., 2010). Using the current much larger dataset of HRV-C sequences and a shorter fragment (positions 616–1004, based on typical coordinates of sequences available in GenBank), pairwise p-distance distributions were plotted, along with those for species A and B (Fig. 6). Detailed inspection of distributions of pairwise distances showed possible assignment thresholds of 10.5, 9.5 and 10.5 % for HRV-A, -B and -C, respectively, based on minimum values. The shorter length and greater sequence conservation than VP1 resulted in less-well-defined thresholds and a greater proportion of pairwise comparisons close to the assignment threshold, which may cause problems with type identification.', 'For example, four type pairs showed VP4/VP2 divergence that was below the proposed threshold [HRV-A31/A47 (0.083), HRV-B52/B104 (0.093), HRV-C4/C49 (0.100) and HRV-C32/C39 (0.090)] even though they were previously assigned as different types based on VP1 divergence and separate clustering by phylogenetic analysis (Table 1 and Fig. 2). More problematic was the larger number of HRV variants of the same type with VP4/VP2 divergence above the assigned thresholds (22, 10 and six types of HRV-A, -B and -C, respectively; Table S3). Most of these types contained several contemporary sequences that had diversified substantially from their corresponding prototype sequence. In these cases, p-distances above the threshold occurred between contemporary isolates diversifying on separate evolutionary lineages, while retaining similarity in the intra-type range with their corresponding prototype type strain [VP4/VP2 distances of <10.5 % (species A) or 9.5 % (species B)]. This is similar to the pattern observed within the VP1 region of contemporary strains of HRV-A29.', 'The original HRV-C classification included 28 PATs, which were temporarily assigned on the basis of VP4/VP2 sequence divergence alone (Simmonds et al., 2010) until a matching VP1 sequence was obtained. Of these original 28 PATS, 16 have since been reassigned as new (confirmed) types based on VP1 divergence (new HRV-C type designations listed in Table S1 and S2), whilst one was combined into an existing type (HRV-C8/Cpat28; see above). VP1 sequences are unavailable for the remaining 11 PATs (Table S4).', 'Eight further divergent VP4/VP2 groups were identified among species A (n = 4), and species B (n = 4) (Table S4). A further three provisionally assigned HRV-C types were also detected (designated HRV-Cpat29-pat31). All showed divergence above the VP4/VP2 threshold proposed for their species. These await sequence data from VP1 to confirm their assignment as different types.', 'The distribution of HRV types (identified using the larger dataset of VP4/VP2 sequences and excluding PATs) was compared for the UK, the rest of Europe, the USA, Asia and other regions (Australia, the Middle East and Africa). Each geographical zone contained similar relative frequencies of HRV-A, -B and -C (Fig. 7). Individual types showed considerable variability in detection frequency, although those that were common showed similar detection frequencies in different geographical areas. Similarly, none of the prevalent HRV types showed any geographical restriction, findings that suggest that most or all HRV types circulate freely worldwide.', 'The ability to perform genotyping directly from clinical specimens removes the need for cell culture passage and therefore greatly reduces the time required for type identification for all three HRV species. It additionally allows variants refractory to virus isolation, such as species C, to be identified. This parallels the transition from isolation and neutralization assay to VP1 sequencing for the identification of EV (sero)types (McWilliam Leitch et al., 2009; Mirand et al., 2009; Nix et al., 2006). Because of its shorter length, greater sequence conservation and the possibility of using the same set of primers for amplification of all three HRV species, the VP4/VP2 region has been most commonly used for genetic characterization of HRV in clinical samples (Blomqvist et al., 2009; Han et al., 2009; Mizuta et al., 2010; Savolainen-Kopra et al., 2009; Wisdom et al., 2009a). Comparison of its predictive value compared with type assignments based on VP1 sequences provides evidence for its suitability of HRV type identification along with some restrictions. The latter include uncertainties in type identifications of variants showing sequence divergences from other HRV sequences close to the threshold assigned to that species. Examples of both inter-type distances below this threshold and intra-type distances above have been presented (Table S3), although patterns of phylogenetic clustering of variants provide further contributory information for type identification.', 'Overall, however, the thresholds we have proposed divide HRV sequences into groups that correspond most consistently with phylogenetic groupings (with the small number of exceptions we have specifically itemised). Placing thresholds in any other position in the distance distributions would create groupings that conflict with those defined by phylogenetic analysis and by serology relationships. Placing the threshold at a minimum point in the distribution additionally minimizes the number of conflicting between- and within-group distances within an assigned type.', 'The identification and definition of HRV types by using genetic data within the capsid region has the potential to revolutionize HRV typing. The ease with which this can be carried out, compared with traditional virus culture and neutralization assays, should make the endeavour of classifying all detected HRV sequences much more feasible and attractive. A more extensive global or sentinel system of type identification should allow large-scale investigations of HRV epidemiology, transmission and evolution; sequence data for comparisons are available worldwide almost instantly and will be of considerable value in surveillance for the emergence of new types of species with differing clinical presentations from those traditionally attributed to HRV.', 'In addition to type identification, the definition of robust nucleotide divergence thresholds for the definition of new EV types (25 %) now provides the primary means for the assignment of new EV types (Oberste et al., 1999a, b). This has been used for several years and has created the assignment of 36 new EV types numbered from EV-73 onwards (Brown et al., 2009; Norder et al., 2003; Oberste et al., 2004, 2007; Smura et al., 2007a, b; Tapparel et al., 2009; Yozwiak et al., 2010). We propose an equivalent system for the assignment of new HRV types in which criteria similar to those proposed for the assignment of species C rhinoviruses (Simmonds et al., 2010) are extended to HRV-A and -B. HRV does, however, differ from EVs in showing lower divergence thresholds for type assignments (12–13 % nucleotide sequence divergence in VP1) and a lack of capsid recombination that enables VP4/VP2 sequences to be used at least for type identification. The criteria for HRV have been formulated as follows:', 'The use of only capsid-coding regions in type-assignment criteria should not detract from the importance of continued investigation of other genomic regions, in particular where these may contribute to the phenotype and disease associations of HRV. In addition, the use of the VP1 region for definition of new HRV types should not discourage the widespread continued use of 5′-untranslated region and VP4/VP2 screening protocols. Screening with these relatively conserved regions allows a much greater opportunity for discovery of previously unknown HRV types through further sequence characterization of VP1 and other genome regions. Although this represents, to our knowledge, the most comprehensive survey of HRV sequence data currently possible, the guidelines should be subject to further review as additional data becomes available.', 'In all three species, the phylogeny of the full VP1 region reliably separated isolates of known differing type into distinct bootstrap-supported clades. This is consistent with the presence of a large number of the neutralizing immunogenic sites in HRV-A and -B (Kistler et al., 2007b) that underlie their serological properties. The difficulties with culturing species C rhinoviruses have to date prevented their serological properties from being directly investigated. It is possible that the sinus mucosal organ culture system used to culture HRV-C15 might prove more broadly applicable for other species C variants (Bochkov et al., 2011). Limited serological reinvestigation of a number of previous HRV-A and -B type assignments would be useful for isolates where cross-reactivity is uncertain (such as HRV-A54/A98) or conflicts with their sequence divergence in VP1.', 'There is a possibility that a smaller fragment of VP1 could be suitable for genotyping in HRV. Oberste et al. (1999b) reported that a 450 nt segment at the 3′ end of VP1 was effective for EV type identification and the results had a 100 % correlation with neutralization results. A second study showed full correlation of a 303 nt stretch of VP1 with neutralization data, but this was only applied to 59 strains (Kiang et al., 2009). In addition, a simple, fast and effective method of typing EVs by pyrosequencing has recently been developed (Silva et al., 2008). This may become extremely useful as pyrosequencing becomes more widely available worldwide. Whether a shorter region than the whole VP1 would be similarly predictive of HRV types and could be used for future type assignments requires further investigation.', 'VP4/VP2 sequences generated from several studies of HRV molecular epidemiology in the UK (Gaunt et al., 2011; Harvala et al., 2012; McIntyre et al., 2010, 2013; Simmonds et al., 2010; Wisdom et al., 2009a, b) (Simmonds et al., unpublished data) (431 HRV-A and 113 HRV-B) were divided into groups based on phylogenetic clustering. Any isolate that did not show a close grouping with a prototype full genome or represented a distinct phylogenetic cluster of HRV-C was selected for amplification of the VP1 region. The procedure for amplification of the VP1 region of HRV-A, -B and -C and initial descriptions of sequences generated have been described previously (McIntyre et al., 2010, 2013; Simmonds et al., 2010).', 'All available HRV VP4/VP2 and VP1 sequences were downloaded from GenBank in November 2012. The genome regions analysed in this study included the full VP1 region and a fragment of VP4/VP2 (nt 616–1004 numbered according to GenBank accession no. EF582385 for HRV-C4) that is commonly used in studies of HRV epidemiology (Blomqvist et al., 2009; Han et al., 2009; Henquell et al., 2012; Kaida et al., 2011; Miller et al., 2009; Mizuta et al., 2010). Any sequences that were <90 % complete across these regions or wereclassified as non-functional were excluded. All sequences were aligned using the sse v1.0 sequence editor package (Simmonds, 2012) and pairwise nucleotide p-distances were calculated using the program Sequence Distances within the package. Phylogenetic trees were constructed in the mega v5.0 software package (Tamura et al., 2011) by the neighbour-joining method (Saitou & Nei, 1987) from 100 bootstrap resampled sequence alignments of maximum composite likelihood (Guindon & Gascuel, 2003) distances, with pairwise deletion for missing data. All phylogenetic analysis was undertaken using the Enterovirus D sequence (GenBank accession no. D00820 for EV-D70) as an outgroup.', 'Phylogenetic trees were inspected visually for evidence of incongruence between the VP4/VP2 and VP1 regions. Datasets containing these two non-consecutive regions were concatenated to give one continuous sequence, which was then additionally analysed for the occurrence of recombination. rdp v4.0 (Martin et al., 2010) analysis was used to screen each HRV species dataset for the occurrence of recombination within the capsid-coding region. Each sequence set was examined using a combination of algorithms implemented in the software package, specifically rdp (Martin & Rybicki, 2000), GeneConv (Padidam et al., 1999), MaxChi (Smith, 1992), Chimaera (Posada & Crandall, 2001), SiScan (Gibbs et al., 2000) and Bootscan (Martin et al., 2005). A threshold P value of <0.05 and detection by at least two algorithms was required for a recombination event to be detected. Each HRV species dataset was further analysed using the gard program (Kosakovsky Pond et al., 2006) as part of the HyPhy package. Any predicted recombination breakpoints were analysed by construction of neighbour-joining maximum composite likelihood trees for sequence fragments 5′ and 3′ to the putative breakpoint.']\n",
            "['All viruses require host translational machinery in order to replicate. To make use of this machinery while avoiding detection, viruses have evolved a number of unconventional translational strategies (Gale et al, 2000), including internal ribosome entry sites (Jang et al, 1988), leaky scanning (Schwartz et al, 1990), ribosome shunting (Schmidt-Puchta et al, 1997), reinitiation (Hemmings-Mieszczak et al, 1997), and programmed ribosomal frameshifting (PRF) (Gesteland and Atkins, 1996). Many of these strategies are unique to viruses, and thus are potential antiviral targets (Harford, 1995).', \"During PRF, the translational machinery slips backward or forward by one nucleotide while decoding a so-called ‘slippery sequence,' a heptanucleotide sequence of the form XXXYYYZ in the mRNA transcript (Figure 1). Outside of PRF, frameshifting is a rare event, occurring fewer than 5 × 10−5 times per codon (Kurland, 1992). PRF can increase the occurrence of frameshifting by several orders of magnitude, depending on the slippery sequence itself as well as on the presence or absence of adjacent secondary structures that favor translational pausing and increase frameshift frequency (Gesteland and Atkins, 1996). In many instances of PRF, slippage causes the translational machinery to bypass a stop codon just downstream of the slippery sequence, resulting in the expression of an elongated protein (Gesteland and Atkins, 1996). Therefore, normal expression from these transcripts results in two products, a shorter product based on the initial translation frame plus a longer product, whose ratio depends on the frequency of slippage and is thought to have evolved to a value optimized for virus production (Shehu-Xhilaga et al, 2001; Dulude et al, 2006).\", 'PRF is an evolutionarily conserved phenomenon observed in a range of viruses, from bacteriophage to notable human pathogens (Table I; Baranov et al, 2002). For example, in HIV, the Gag to Gag-Pol expression ratio is regulated by PRF and is necessary for the proper loading of reverse transcriptase into the new viral capsids; disturbing this ratio can inhibit viral replication (Shehu-Xhilaga et al, 2001; Dulude et al, 2006). The role of PRF in the SARS life cycle is less clear, but is necessary for expression of its RNA-dependent RNA polymerase (Plant et al, 2010). In Escherichia coli bacteriophage lambda, PRF regulates the expression of proteins gpG and gpGT (Levin et al, 1993). As lambda phage genes G and T lie between its major tail gene (V) and its tape measure gene (H), they are thought to have a role in tail formation (Xu, 2001). However, proteins gpG and gpGT do not make up any of the final tail structure (Levin et al, 1993), and thus may act as chaperones or scaffolds for gpH, assisting in the assembly of the tail shaft around the tape measure protein (Xu, 2001).', \"Reading-frame maintenance is a common function of tRNA modifications (Urbonavicius et al, 2001). tRNAs contain upwards of 80 known modified nucleosides (Limbach et al, 1994), and many of these are strongly conserved across species (Sprinzl and Vassilenko, 2005). However, their loss is generally not lethal (Alexandrov et al, 2006; Phizicky and Hopper, 2010). In E. coli, four tRNA modifications require the addition of a thiol moiety and are made via two functionally distinct mechanisms (Nilsson et al, 2002). The first mechanism requires proteins that contain iron-sulfur (Fe-S) clusters and includes 2-thiocytidine formation at position 32 and methylthio formation of ms2i6A at position 37 (Jager et al, 2004; Pierrel et al, 2004). The second mechanism is independent of Fe-S clusters and includes 4-thiouridine at position 8 and 2-thiouridine at position 34 (Ikeuchi et al, 2006). For each of these modifications, sulfur is sequestered by E. coli's primary cysteine desulfurase, IscS. Sulfur is then used for one of these four tRNA modifications or for Fe-S cluster biosynthesis. In addition to their roles in tRNA modification, Fe-S clusters are critical cofactors involved in many cellular processes including respiration, central metabolism, environmental sensing, RNA modification, DNA repair, and DNA replication (Py and Barras, 2010).\", 'We previously identified two pathways downstream of IscS in a screen to identify E. coli genes with a significant effect on lambda phage replication (Maynard et al, 2010). Specifically, lambda phage replication was inhibited following deletion of several members of the 2-thiouridine synthesis (TUS) pathway leading to 2-thiouridine modification of tRNALys/Glu/Gln, a pathway independent of Fe-S cluster biosynthesis. Conversely, we found that knocking out members of the Fe-S cluster biosynthesis (ISC) pathway enhanced lambda replication.', \"The current investigation was therefore motivated by this overarching question: how do the host's TUS and ISC pathways control viral replication? We were particularly intrigued by the possibility that these pathways interact. Furthermore, the substantial evolutionary conservation of both host genes and viral PRF mechanism strongly suggested that our observations would have relevance beyond E. coli and lambda phage. Our current observations shed light on a complex network that extends from host metabolism and tRNA modification to viral translational regulation and finally to virion production. We find that 2-thiouridine hypomodification of tRNALys/Glu/Gln causes increased translational frameshifting, changing the ratio of the critical lambda phage proteins gpG and gpGT. We also show that IscU is linked to gpG and gpGT expression by competitive inhibition of the TUS pathway. Knocking out core members of the ISC pathway increases lambda phage replication by relieving competition for sulfur, allowing the TUS pathway to increase the rates of 2-thiouridine formation, thus reducing frameshifting. Targeting tRNA modifications to alter frameshifting rates, to which viral structural proteins are likely to be uniquely sensitive, presents a novel antiviral strategy.\", 'In a previous genome-wide screen (Maynard et al, 2010), we found that plates growing TUS pathway deletion strains (ΔtusA, ΔtusE, and ΔmnmA) infected with lambda phage produced plaques of an unusually small diameter compared with wild-type (WT) E. coli. In contrast, several ISC pathway deletion strains (ΔiscU, ΔhscA, and ΔhscB) produced abnormally large diameter plaques compared with WT.', 'We investigated viral replication in these strains more thoroughly by culturing each strain in liquid media, in the presence and absence of virus. Without phage, WT E. coli grew in exponential phase for several hours, slowing and reaching stationary phase as available nutrients decreased (Figure 2A, blue line). In the presence of lambda phage, the culture exhibited three phases (gray line). First, the culture underwent exponential growth similar to the uninfected culture. After ∼6 h in the WT strain, viral lysis overtook the culture and the culture began to clear (lytic phase). Lambda phage is a temperate phage with both lytic and lysogenic life cycles (Landy and Ross, 1977); as a result, after a roughly 3-h lytic phase, lysogenized bacterial population growth took over the culture.', 'The time courses of lambda phage infection of the tusA and iscU knockout strains (the first members of the TUS and ISC pathways, respectively) differed significantly from that of WT. The ΔiscU culture entered the lytic phase very quickly (red curve, Figure 2B), while the infected ΔtusA strain grew in exponential phase substantially longer than the WT strain, leading to a higher turbidity before the lytic phase occurred (green curve). We therefore decided to classify strains that entered the lytic phase earlier as more susceptible to viral infection (red) and those that entered the lytic phase later as less susceptible (green). This susceptibility classification was consistent with our earlier plaque assay experiments (Maynard et al, 2010).', 'We then defined a metric to compare the resistance of different strains with infection with phage lambda. One useful way to summarize the information contained in a time course is to consider each experiment as a vector in n-dimensional space, where n is the number of time points taken. In earlier work, we found that it was useful to normalize these infection time course experiment vectors by the growth rate (Maynard et al, 2010). However, we have since found that bacterial growth and viral growth are tightly linked, and our attempts to numerically correct for or eliminate the effect of bacterial growth rate often distorted the raw data. Bacterial growth is therefore an important (but not sufficient) consideration in classifying our strains.', \"As a result, we determined our new metric simply by calculating the Euclidean distance between the knockout and WT strain time course vectors under infection conditions. Combining this distance with our susceptibility classification, we obtain a displacement vector that points in a negative direction for strains that are less resistant to virus and a positive direction for more resistant strains. The displacement vector, which we call the ‘host resistance', enables us to compare the magnitude and direction of each strain relative with the WT strain and with each other. For example, the ΔiscU strain is significantly less resistant to viral infection than the WT strain, while the ΔtusA strain is more resistant (Figure 2C).\", 'As both the TUS and ISC pathways are linked to sulfur metabolism through IscS, we wondered about the interplay between these two pathways and what might be the mechanistic basis for their effect on lambda replication. We began our investigation by examining the impact of other known ISC and TUS pathway members on lambda phage replication (Figure 3A; see Supplementary Figure 1 for line plots of individual replicates). First, we investigated the effect on lambda phage replication of several genes associated with Fe-S cluster biosynthesis. We performed infection time courses of ΔiscU and other strains deficient in members of the iscRSUA-hscBA-fdx-iscX operon (hscA, hscB, iscA, iscR, fdx, and iscX). The slow growth of the iscS deletion strain made it impossible to assess infection dynamics and was not included in the analysis. The ΔiscU, ΔhscA, and ΔhscB strains all exhibited very similar infection dynamics, clearing the culture significantly before WT (Figure 3A, top row; Figure 3B). IscU is the primary scaffolding protein used for Fe-S cluster biosynthesis (Urbina et al, 2001), and HscA and HscB are accessory proteins that help transfer Fe-S clusters to apoproteins (Bonomi et al, 2008) The ΔiscA strain exhibited increased replication efficiency, but less than the hscA, hscB, and iscU gene deletions (Figure 3A and B). IscA assists in Fe-S cluster biosynthesis by shuttling iron to IscU (Yang et al, 2006).', 'Deletion strains for the remaining three genes in the operon, iscR, fdx, and iscX, displayed little if any change in viral infection dynamics from WT (Figure 3A and B). The roles of Fdx and IscX in Fe-S cluster biosynthesis are not known (Takahashi and Nakamura, 1999), but IscR is the transcriptional repressor of the ISC operon and its active form includes an Fe-S cluster (Schwartz et al, 2001). Interestingly, lambda phage replication is slightly but significantly hindered in the iscR knockout strain (Figure 3A and B). Taken together, these observations suggest that ISC pathway members IscU, HscA, HscB, IscA, and possibly IscR have a more significant role in lambda phage infection than Fdx or IscX, and that their WT function helps to restrict phage growth.', 'We next determined how deletion of tRNA thiolation enzymes affected viral susceptibility of the host. We began with the Fe-S cluster-containing enzymes encoded by miaB and ttcA. MiaB is responsible for methylthio formation of ms2i6A at position 37 in tRNAPhe (Pierrel et al, 2004); a sulfur donor for MiaB has not been identified. TtcA catalyzes the 2-thiocytidine modification at position 32 of tRNAArg(ICG), tRNAArg(CCG), tRNAArg(mnm5UCU), and tRNASer(CGU) (Jager et al, 2004). ΔttcA strains displayed doubling times equivalent to WT E. coli (Figure 3A, bottom), and little is known about the functional consequences of modification. Strains lacking either of these genes showed no effect on lambda infection dynamics (Figure 3A, bottom; Figure 3C). In contrast, deletion strains for the two Fe-S cluster-independent tRNA thiolation enzymes ThiI and MnmA displayed a significant increase in viral resistance versus the WT strain (Figure 3A, bottom; Figure 3C). Sulfur is transferred to ThiI through direct binding of IscS (Kambampati and Lauhon, 2000), and ThiI then transfers the sulfur to s4U8 of tRNAPhe. Lambda replication in the ΔthiI strain was mildly but significantly inhibited (Figure 3A, bottom; Figure 3C).', 'The ΔmnmA strain exhibited the most dramatic effect on viral infection of all the strains associated with tRNA thiolation. MnmA catalyzes 2-thiouridination at position 34 of several tRNAs (Kambampati and Lauhon, 2003). Sulfur is supplied to MnmA through the TUS pathway, beginning with the direct binding of IscS by TusA (Ikeuchi et al, 2006). The ΔmnmA strain underwent infection dynamics (Figure 3A, bottom; Figure 3C) very similar to the upstream TUS pathway gene deletion strains we characterized in a previous study (Maynard et al, 2010), including ΔtusA, ΔtusBCD, and ΔtusE. All members of the TUS pathway had a strong, significant, and similar effect on lambda replication (Figure 3A and C), leading us to conclude that Fe-S cluster-independent tRNA thiolation in E. coli exerts a significant effect on lambda phage replication through the TUS pathway and, to a lesser degree, ThiI.', \"Based on the preceding data and a previous observation of increased frameshifting in ΔmnmA strains (Urbonavicius et al, 2001), we hypothesized that reduced lambda replication in the TUS knockouts was due to increased frameshifting during lambda protein synthesis as a result of the decreased tRNA modification. To test this hypothesis, we generated infection curves for several deletion mutants that have been shown to affect frameshifting by loss of tRNA modification, including tgt, truA, miaA, thiI, and mnmE (Urbonavicius et al, 2001). We infected deletion strains for these genes with lambda phage and determined their susceptibility to infection (Figure 4; see Supplementary Figure 2 for line plots of individual replicates). Δtgt, ΔmiaA, and ΔtruA exhibited no significant affect on lambda phage replication, suggesting that the tRNA modifications from these enzymes have no role in lambda phage replication. The remaining knockout strains displayed significant changes in lambda phage's ability to infect the host, including ΔthiI, ΔmnmA (Figures 3 and 4), and ΔmnmE (Figure 4). Intriguingly, MnmE performs a methylaminomethyl modification on carbon 5 of the same tRNA species and nucleoside targeted by MnmA (Elseviers et al, 1984), suggesting that the antiviral properties of the TUS pathway knockouts may be due to specific tRNA modifications instead of to general translational fidelity issues.\", 'Host-preferred codon usage is common among many viruses and their hosts (Kunisawa et al, 1998; Lucks et al, 2008). We wondered whether the lambda genome used the codons whose tRNAs are modified by the mnmA, and mnmE gene products preferentially with respect to E. coli. The tRNA species modified by MnmA and MnmE decode the lysine, glutamic acid, and glutamine codons (Kambampati and Lauhon, 2003); we examined the codon usage for lysine, glutamic acid, and glutamine in lambda phage and in E. coli for synonymous codon usage bias. Lambda phage encodes lysine 748 times (65% AAA and 35% AAG), glutamic acid 900 times (57% GAA and 43% GAG), and glutamine 576 times (20% CAA and 80% CAG), with similar biases in E. coli (Table II). This similarity in codon usage does not support the hypothesis that lambda phage protein synthesis is any more impaired in strains harboring hypomodified 2-thiouridine tRNAs than its host.', 'Since we thought it unlikely that lambda phage replication was being significantly affected by general reading frame maintenance in the TUS pathway knockout mutants, we turned instead to specific tRNA modifications that could explain our observations of an increase in viral resistance. In particular, our attention was drawn to PRF as a specific instance of frameshifting that could be particularly sensitive to changes in frameshifting frequency.', 'In lambda phage, PRF occurs during translation of the lambda phage GT region, which encodes two possible protein products, either gpG or gpGT; expression of gpGT depends on ribosomal frameshifting at the slippery sequence GGGTTTG (Levin et al, 1993). This sequence encodes the dipeptide Gly-Lys in both the 0 open reading frame (ORF; GGT-TTG) and the −1 ORF (GGG-TTT). In the 0 ORF, which is maintained ∼96% of the time, gpG is expressed (Figure 5A); and in the remaining cases, the ribosomal machinery reaches the slippery sequence and slips back one base pair. As a result, the stop codon for gpG, immediately downstream of the slippery sequence, is bypassed and the larger gpGT product is produced. Interestingly, both the lysine codons TTG (0 ORF) and TTT (−1 ORF) are decoded by the same tRNALys (UUU), a target of MnmA and MnmE. We therefore wondered whether TUS pathway gene deletion strains would exhibit an altered rate of frameshifting and consequently a change in the ratio of gpG to gpGT.', 'To monitor gpG and gpGT expression in E. coli, we created a plasmid for the inducible synthesis of these proteins (pBAD-λGT; Figure 5B). We PCR amplified the GT region of the lambda genome and introduced this region behind an arabinose-inducible promoter, to control expression of the transcript, and the Xpress Epitope C-terminal tag, to facilitate detection by immunoblotting. We then transformed WT, ΔtusA, ΔiscU, and ΔtusAΔiscU strains with pBAD-λGT. Deletion of the iscU gene led to abrogation of frameshifitng, whereas frameshifting frequency increased in both the ΔtusA and the ΔtusAΔiscU strains relative to WT (Figure 5C). We also tested frameshifting frequency for additional strains (ΔmnmA, ΔthiI, ΔmiaB, ΔttcA, ΔhscA, ΔhscB, Δfdx, ΔiscA, and ΔiscR) and found frameshifting levels to be consistent with infection dynamics (see Supplementary Figure 3). We interpret these results to signify that the decreased thiolation of tRNALys (UUU) in the TUS pathway knockout strains leads to an increase in frameshifting at the slippery sequence.', 'Surprisingly, the ΔiscU strain exhibited markedly reduced gpGT levels relative to the WT strain (Figure 5C), strongly indicating that both IscU and TusA act on lambda phage replication by altering the gpG:gpGT ratio. Furthermore, our observation that the miaB and ttcA deletion strains exhibited normal lambda phage replication (Figure 3) suggested that IscU does not act directly through thiolation of specific tRNA modifications. How, then, does IscU influence viral replication independent of direct tRNA modification?', 'One possible answer is that IscU competes with TusA to bind IscS. Recent X-ray crystallography studies showed that IscU and TusA bind IscS at distinct but adjacent locations (Shi et al, 2010). Superposition of the IscU and TusA structures indicated a spatial overlap between the volume occupied by TusA and IscU when bound to IscS. Furthermore, three-way pull-down experiments demonstrated that IscS may bind either TusA or IscU individually, but not both at the same time (Shi et al, 2010). To further explore the possibility that thiolation of tRNALys (UUU) is affected by competition between TusA and IscU for sulfur transfer from IscS, we constructed a mathematical model of IscU and TusA binding to IscS (Figure 6A). Our model consisted of a set of equilibrium and mass conservation relationships. In this case, the model parameters had either been previously measured or were not difficult to estimate based on existing data (Urbina et al, 2001; Taniguchi et al, 2010). Our model predicts the relative amount of U34 thiolation from the TUS pathway for a specified amount of IscU.', \"We combined this protein binding model with our previous mathematical model of E. coli infection by lambda phage (Maynard et al, 2010). Our earlier model predicts the E. coli time course dynamics of lambda phage infection based on three parameters: the fraction of infections that are lytic instead of lysogenic, the rate of infection, and the burst size (how many functional progeny are released upon lysis). To integrate these two models, we related U34 thiolation (the output of the protein binding model) to the burst size (input for our infection model). We were thus able to explore the effect of changing the IscU or TusA concentration on the production of functional phage and the resulting infection time course. We first tested the integrated model's ability to predict the outcome of deleting iscU or tusA, and in both cases the model was able to capture the infection time course (Figure 6B–E; see Supplementary Figure 4 for line plots of individual replicates). We also considered titration of IscU experimentally as well as computationally by introducing an IPTG-inducible iscU gene construct into a ΔiscU strain and determining the infection time courses under various IPTG concentrations. The effect of IscU titration was predicted well by our competitive inhibition model (Figure 6F and G; see Supplementary Figure 4 for line plots of individual replicates).\", 'Finally, we considered the case of a double deletion of tusA and iscU. The competitive inhibition model suggested that the double mutant would behave in a manner similar to a ΔtusA strain under infection conditions (Figure 6H). Again, this prediction agreed with the experimental time course (Figure 6I and J; see Supplementary Figure 4 for line plots of individual replicates), as well as our experimental observation that the ratio of gpGT to gpG in the double mutant was similar to that observed in the ΔtusA strain, but not in the ΔiscU strain (Figure 5C).', 'As a control, we compared the predictions of the competitive inhibition model with those of the alternative hypothesis that IscU and TusA influence viral infection independently. This hypothesis required some relatively simple modifications in the equations and parameters of our competitive binding model (Materials and methods). Our independent effect model was able to capture the results of tusA deletion, iscU deletion, and IscU titration as accurately as the competitive inhibition model (data not shown). However, the independent effect model was not able to predict the behavior of the double knockout strain (Figure 6K).', \"Our competitive inhibition model predicts that the removal of IscU, one of TusA's primary competitors for sulfur, will cause an increase in the fraction of 2-thiolation modified U34 tRNALys(UUU) relative to BW25113. To directly measure the fraction of thiolated tRNALys(UUU), we performed an [(N-Acryloylamino)phenyl] mercuric chloride (APM) northern blot (Figure 7), which specifically retards the migration of the thiolated tRNA fraction (Igloi, 1988). We found a small fraction of hypomodified tRNALys in BW25113, and as predicted by the competitive inhibition model, loss of IscU in the ΔiscU strain decreased this fraction even further. We therefore conclude that our competitive inhibition model, but not our independent effect model, is sufficient to explain the effect of TUS and ISC pathway knockouts on lambda phage replication.\", 'Taken together, our observations suggest the existence of a complex network that links sulfur metabolism, tRNA modification, competitive binding, and PRF-based regulation of viral protein expression with host susceptibility to viral infection (Figure 8). TusA obtains sulfur from the cysteine desulfurase IscS, and passes it along the TUS pathway for eventual modification of tRNALys(UUU) at U34. The fraction of tRNALys(UUU) that undergoes this modification influences the frequency of PRF in lambda phage due to its role in decoding part of the slippery sequence within its GT region. Although lambda phage requires both gpG and gpGT for normal replication, the gpG:gpGT ratio must be kept high. Deletion of any member of the TUS pathway prevents modification of tRNALys(UUU), and subsequently increases frameshifting, both decreasing the gpG:gpGT ratio and lambda phage production.', 'In the presence of HscA, HscB, and (less critically) IscA, IscU binds IscS more strongly than TusA (Shi et al, 2010) and is therefore able to outcompete TusA for sulfur. When IscU, HscA, HscB, or IscA is removed from the system, TusA obtains more sulfur than normal, leading to hypermodification of tRNALys(UUU), a decrease in frameshifting, and increases in the gpG:gpGT ratio and lambda phage production. Previous investigations in B. subtilis reported that disrupting the function of IscU affects several levels of tRNA modifications, including an increase in 2-thiolation of U34 (Leipuviene et al, 2004); our observations suggest that in the case of tRNALys(UUU), this effect is a result of an increase of sulfur flux through the TUS pathway (Figures 5C, 6K, and 7). Removal of the transcriptional repressor IscR leads to an increase in protein production from the ISC operon (Schwartz et al, 2001), reducing the amount of sulfur available to TusA. This competition effect depends on an intact TUS pathway, which is why the ΔtusAΔiscU deletion mutant expresses a phenotype similar to the ΔtusA strain.', \"This model fits our observations well, but unresolved issues remain, most notably the role of ThiI in viral infection. ThiI would seem to be another possible competitor for sulfur from IscS (Figure 2A). However, the ΔthiI strain exhibited decreased lambda replication, albeit somewhat less strongly than the TUS pathway deletion strains. One clue to this mystery may reside with the CyaY protein, which is thought to be an iron donor for Fe-S cluster assembly, forming a ternary complex with IscU and IscS (Shi et al, 2010). The binding site for ThiI on IscS overlaps substantially with CyaY (Shi et al, 2010), and it is conceivable that in the absence of ThiI, CyaY is able to interact more efficiently with IscU, either stabilizing the IscS-IscU-CyaY complex or assisting sulfur transfer to IscU. In either case, deleting thiI may increase IscU's ability to compete with TusA for sulfur.\", 'Our findings point to several novel antiviral strategies and targets. Although this work was performed using E. coli and phage lambda, many of the key elements of the network we describe are conserved in other systems, in both in host and the virus. Both Fe-S clusters and tRNA modifications are ubiquitous in cells (Limbach et al, 1994; Py and Barras, 2010) and PRF is a common viral translational strategy (Gesteland and Atkins, 1996). Indeed, frameshifting has already been identified as a potential target in HIV, but efforts to target the gag-pol mRNA secondary structure have been challenging (Hung et al, 1998). Our results point to upstream (e.g., tRNA modifying enzymes) and even indirect targets (e.g., Fe-S cluster biosynthesis pathways), which can just as potently alter infection and in some cases are tunable (Figure 6G). We therefore anticipate that our observations and inferences may have broad relevance to understanding other infections, including viral infection of humans.', 'The E. coli strains used in this study were all taken or derived from the Keio Collection, a comprehensive library of single gene deletion strains (Baba et al, 2006). BW25113 is the parental strain used for the construction of the Keio Collection. Lambda phage was obtained from the ATCC (23724-B2). The ΔtusBCD strain was generated previously (Maynard et al, 2010), and the ΔtusAΔiscU double knockout strain was constructed according to the Datsenko-Wanner method (Datsenko and Wanner, 2000). Primer sequences are available upon request. The lambda GT reporter strain was constructed using a pBAD/His A plasmid (Invitrogen, Cat# V430-01). The lambda phage GT region was amplified via PCR (primers: 5′-cacattctcgagatgttcctgaaaaccgaatca-3′ and 5′-gtgtaaaagcttgccataccggactcctcct-3′). The amplified GT region and pBAD/His A plasmid were cut with XhoI and HindIII. BW25113, ΔtusA, ΔiscU, and ΔtusAΔiscU strains were transformed with the arabinose-inducible lambda phage GT reporter construct (Figure 5B). A chemically inducible ΔiscU strain was created by transforming ΔiscU with pCA24N-iscU isolated from the ASKA collection (Kitagawa et al, 2005).', 'We monitored infection dynamics using an incubated plate reader (Perkin-Elmer Victor3, 2030-0030) under growth conditions described previously (Maynard et al, 2010). Exponentially growing cells were normalized to 0.1 OD 600 nm. Fifteen microliters of 0.1 OD E. coli and 15 μl of ∼104 plaque forming units (p.f.u.)/ml lambda stock were added to 170 μl of LB medium in 96-well plates, representing a multiplicity of infection of ∼2 × 10−4 p.f.u./bacteria. Each plate contained four technical replicates of infected and uninfected samples. The plate reader measured the absorbance of the culture approximately every 17 min for 16 h. Each plate contained a sample of BW25113 for between-plate normalization. Three independent biological replicates were performed for each strain.', 'Time courses were compared based on a useful displacement metric that includes a direction in addition to a distance. The distance was determined by calculating the Euclidean distance between the two time course vectors; the Euclidean distance between the recorded absorbances for each culture were calculated at each time point, and these distances were summed for all time points. For our purposes, the direction could only hold two values, +/−1. The positive value was associated with strains exhibiting increased resistance to viral infection, and the negative value was associated with increased susceptibility to viral infection. The direction was determined by the difference in the time it took a culture to reach its peak absorbance. For example, time courses with longer times to peak compared with BW25113 were considered to be displaced in a positive direction from BW25113. The displacement was then calculated as (direction) × (distance).', 'The Euclidean distance from BW25113 was measured for each biological replicate. Statistical significance was determined using a two-sample t-test comparing the Euclidian distances calculated for each biological replicate and its distance from BW25113 for each plate containing the strain being compared.', 'Lambda phage ORFs were extracted from NCBI Reference Sequence NC_001416.1 (Enterobacteria phage lambda, complete genome) and codon usage was tallied.', 'Codon usage for WT E. coli (W3110) was determined from the Codon Usage Database (http://www.kazusa.or.jp/codon/).', 'pBAD-λGT-transformed cells were cultured overnight at 37°C in LB and the appropriate antiobiotics. The cultures were diluted 1:50 into fresh LB the following morning and placed back in the shaker for 1 h. The culture was then split into 1 ml aliquots arabinose. Arabinose induction was performed for 2 h at 0.2, 0.02, and 0.002% arabinose, after which the samples were spun down in a table-top centrifuge at ∼20 000 g for 30 s. The media was aspirated and the pellets were frozen at −80°C. The samples were run on a 12% SDS–PAGE gel for 1.5 h at 80 V in Tris/glycine/SDS buffer. The protein was transferred onto a polyvinylidene fluoride membrane for 1 h at 100 V at room temperature. The membrane was washed for 30 s in TBST, then blocked with 5% milk blocking buffer for 1 h. The primary antibody (Anti-express antibody, Invitrogen) was diluted 1:5000 and incubated with the membrane blot overnight at 4°C. The membrane was washed five times for 10 min with TBST, followed by a 1-h incubation with a 1:5000 dilution of the secondary antibody (goat anti-mouse IgG horseradish peroxidase). The membrane was washed five times for 5 min with TBST. Luminol reagent was used for chemiluminescence imaging (Santa Cruz Biotechnology, sc2048).', 'BW25113 and ΔiscU were cultured overnight and diluted 1:100 to 20 ml in the morning. When the cultures reached ∼0.5 OD 600 nm, the samples were placed on ice. The cells were pelleted at 5000 r.p.m. for 5 min at 4°C and the supernatant was decanted. tRNA enrichment was performed using standard techniques. Briefly, the cell pellets were resuspended in 300 μl cold sodium acetate (0.3 M, pH 4.5, 10 mM EDTA) and transferred to eppendorf tubes. The samples were snap frozen in liquid nitrogen. After thawing, 300 μl phenol:chloroform (pH 4.7) was added to the samples and vortexed four times for 30 s bursts with 1 min incubation on ice in between. The samples were centrifuged for 15 min at 4°C in a table-top centrifuge at maximum speed. The top layer (aqueous phase) was transferred to a new tube and the acid phenol extraction procedure was repeated. After the aqueous phase was transferred to a new tube, the RNA was ethanol precipitated by adding three volumes of cold 100% ethanol and spinning at maximum speed for 25 min at 4°C. The RNA pellet was resuspended in 60 μl cold sodium acetate (0.3 M, pH 4.5) and precipitated again by adding 400 μl 100% ethanol and spinning at maximum speed for 25 min at 4°C. The supernatant was decanted and all traces of ethanol were removed. The RNA pellet was air dried on ice then resuspended in 100 μl cold sodium acetate (10 mM, pH 4.5).', \"The APM northern blot was performed according to the protocol specified by Igloi (1988). Briefly, two 8 M Urea, 5% acrylamide gels were cast in 0.5 × TBE buffer. One gel contained 200 μl of 1 mg/ml APM (kindly supplied by Gabor Igloi) for 10 ml of gel. The gels were prerun for 30 min at 180 V in 0.5 × TBE. Approximately 2.5 μg of RNA (for each well) was then mixed with 2 × loading buffer and heated to 65°C for 3 min before being cooled on ice for 1 min. The gels were run for ∼45 min at 180 V. The RNA was transferred onto a nylon membrane using a semi-dry system for 1 h at 250 mA. The transfer buffer contained 0.5 × TBE supplemented with 10 mM β-mercaptoethanol. The RNA was then UV crosslinked to the nylon membrane. T4 Polynucleotide Kinase (PNK) was used to radiolabel 25 pmol of 32P ATP to 25 pmol of tRNALys(UUU) oligonucleotide probe for each blot. The oligonucleotide sequence used to probe tRNALys(UUU) was 5′-TGGGTCGTGCAGGATTCGAA-3′. Qiagen's Qiaquick Nucleotide Exchange Kit was used to remove free 32P ATP. The blots were incubated in 25 ml of prehybridization buffer (10 × Denhardt's, 6 × SSC, 0.1% SDS) at 42°C for 1 h. The labeled probe was heated to 95°C for 2–3 min before being added to the prehybridaztion buffer. The blots were hybridized at 42°C overnight. The next day, the blots were washed with high salt buffer (5 × SSC, 0.1% SDS) at 37°C for 20 min twice, followed by two washes with low salt buffer (1 × SSC, 0.1% SDS) at 42°C for 20 min. After the final wash, the blots were exposed to a phosphorscreen for 10 min.\", 'We constructed a competitive inhibition model based on the following assumed mechanisms: (1) the cysteine desulfurase IscS binds sulfur, converting cysteine to alanine in the process; (2) thiolated IscS presents sulfur to the apo forms of TusA and IscU; (3) TusA and IscU bind thiolated IscS with their respective affinities and obtain the sulfur; (4) TusA and IscU transfer the sulfur to their downstream pathways, represented as sTUS and sISC, respectively. The source code for our model can be downloaded from https://simtk.org/home/lambda-tus.', 'The reactions are as follows:', '', 'From these reactions we derived the following set of ordinary differential equations and mass conservation relationships:', '', 'Many of the parameter values were taken from the literature. Taniguchi et al (2010) reported values of ∼800 IscS proteins per cell and 450 IscU proteins per cell. Approximating the E. coli volume as 10−15 L, we obtained concentrations of 1.37 and 0.77 μM for IscS and IscU, respectively. Since TusA amounts were not reported, we estimated its concentration to be similar to but less than IscU 0.5 μM. An in-vitro study reported the Kd values of IscS and IscU to be 2 μM, and the kcat (k2 in the above equations) to be 8.5 per minute (Urbina et al, 2001). IscU displaced TusA in a three-way pull-down assay (Shi et al, 2010), from which we hypothesized that the Kd of TusA-IscU was 20% more than that of IscU-IscS. The cysteine concentration was chosen to be equal to the IscS concentration.', 'The forward rates of reaction were assigned as follows: k1=k3=k5=k7=k8=105, k4=k6=k−3. The reverse reaction rates were obtained by multiplying the corresponding forward rate by the appropriate dissociation constant. Gene deletion conditions were simulated by setting the pertinent concentration values to zero. IscU titrations were modeled by setting its concentration to 50, 100, and 150% of the WT amount.', 'Steady-state values of the sTUS parameter relative to WT were used to scale the amplification factor b in our previous non-dimensionalized predator-prey model (Maynard et al, 2010). We simulated infection dynamics with the following parameters: , μ*=0.3, K*=0.4, ks=0.', 'The independent effect model hypothesizes that there is no competition between IscU and TusA for sulfur. Thus, they each receive their own (equally sized) pool of thiolated IscS (iscSU·S belongs to IscU and iscSA·S belongs to TusA). The dissociation constants of both IscU and TusA with their respective pools of IscS were set to 2.7 μM. The concentrations of IscU and TusA were set to the same values as in the competitive model. The amount of thiolated IscS, however, was increased 100-fold to prevent it from being limiting. The reactions are as follows:', '', 'From these, we derived the following set of ordinary differential equations and constraints:', '', 'The kinetic parameters were specified as follows: k1f=k2f=k3f=k4f=105. Non-zero reverse rates were set to the product of the forward rate and the appropriate dissociation constant.', 'Steady-state values of the sTUS and sISC parameters relative to WT were used to scale the amplification factor b. Parameters for the predator-prey equation were . We see that in both models, the WT burst rate is 20 and the tusA knockout burst rate is 10. Supplementary Table 1 describes the parameters used in the two models.', 'The standard ode23s solver in Matlab was used for solving the sulfur transfer equations. The ode45 solver was used to solve the predator-prey equations.']\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "OxKiABSBMH2p"
      },
      "source": [
        "def preprocessing(text):\r\n",
        "    # remove mail\r\n",
        "    text = re.sub(r'[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,}', 'MAIL', text)\r\n",
        "    # remove doi\r\n",
        "    text = re.sub(r'https\\:\\/\\/doi\\.org[^\\s]+', 'DOI', text)\r\n",
        "    # remove https\r\n",
        "    text = re.sub(r'(\\()?\\s?http(s)?\\:\\/\\/[^\\)]+(\\))?', '\\g<1>LINK\\g<3>', text)\r\n",
        "    # remove single characters repeated at least 3 times for spacing error (e.g. s u m m a r y)\r\n",
        "    text = re.sub(r'(\\w\\s+){3,}', ' ', text)\r\n",
        "    # replace tags (e.g. [3] [4] [5]) with whitespace\r\n",
        "    text = re.sub(r'(\\[\\d+\\]\\,?\\s?){3,}(\\.|\\,)?', ' \\g<2>', text)\r\n",
        "    # replace tags (e.g. [3, 4, 5]) with whitespace\r\n",
        "    text = re.sub(r'\\[[\\d\\,\\s]+\\]', ' ',text)\r\n",
        "     # replace tags (e.g. (NUM1) repeated at least 3 times with whitespace\r\n",
        "    text = re.sub(r'(\\(\\d+\\)\\s){3,}', ' ',text)\r\n",
        "    # replace '1.3' with '1,3' (we need it for split later)\r\n",
        "    text = re.sub(r'(\\d+)\\.(\\d+)', '\\g<1>,\\g<2>', text)\r\n",
        "    # remove all full stops as abbreviations (e.g. i.e. cit. and so on)\r\n",
        "    text = re.sub(r'\\.(\\s)?([^A-Z\\s])', ' \\g<1>\\g<2>', text)\r\n",
        "    # correctly spacing the tokens\r\n",
        "    text = re.sub(r' {2,}', ' ', text)\r\n",
        "    text = re.sub(r'\\.{2,}', '.', text)\r\n",
        "    text=re.sub(r\"[<>()(,)|&©ø\\[\\]\\'\\\";?~*!]\", ' ', text) #remove <>()|&©ø\"',;?~*!\r\n",
        "    text=re.sub(\"(\\\\t)\", ' ', text) #remove escape charecters\r\n",
        "    text=re.sub(\"(\\\\r)\", ' ', text) \r\n",
        "    text=re.sub(\"(\\\\n)\", ' ', text)\r\n",
        "    text= re.sub(\"(\\s+)\",' ',text) #remove multiple space\r\n",
        "    # return lowercase text\r\n",
        "    return text.lower()"
      ],
      "execution_count": 83,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "tle9VT35P1fU"
      },
      "source": [
        "from nltk.tokenize import sent_tokenize\r\n",
        "full_body_text2=[]"
      ],
      "execution_count": 127,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "56NVitVlMtKJ"
      },
      "source": [
        "for i in full_body_text :\r\n",
        "  txt = preprocessing(i[0])\r\n",
        "  t=sent_tokenize(txt)\r\n",
        "  full_body_text2.append(t)"
      ],
      "execution_count": 128,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1aDfBtX9QOq4",
        "outputId": "3fb4eb3e-43c7-45d3-ea4f-41f994688522"
      },
      "source": [
        "for s in full_body_text2 :\r\n",
        "  print(s)\r\n",
        "len(full_body_text2) "
      ],
      "execution_count": 130,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "['emerging and re-emerging viral infections are becoming severe global public health problems in the current century.', 'in february 2003 an outbreak of severe acute respiratory syndrome sars caused by sars coronavirus sars-cov was reported in guangdong china ksiazek et al 2003 .', 'the sars-cov infected 8096 cases and 774 deaths worldwide.', 'in march 2009 a novel influenza virus h1n1pdm emerged in the united states and mexico.', 'h1n1pdm obtained the capacity to transmit in humans and quickly spread to more than 214 countries hendrickson and matthay 2013 .', 'thereafter the h1n1 infection became a seasonal virus circulating over the world dawood et al 2009 .', 'middle east respiratory syndrome mers coronavirus was isolated from patients who developed acute pneumonia and renal failure in 2012 from saudi arabia zaki et al 2012 .', 'one big secondary outbreak with 186 confirmed cases was in south korea in 2015. up to january 2020 more than 2500 confirmed cases were reported with a case fatality of 34 4% who 2019 .', 'in december 2019 a novel coronavirus sars-cov-2 caused an outbreak of acute pneumonia in wuhan city of china zhou p et al 2020 .the current animal models for study of sars-covs include african green monkeys macaques ferret and mice kong et al 2009 smits et al 2011 .', 'these in vivo models are useful for studying many important questions for example local or systemic pathogenic changes immune response and drug metabolism and in vivo efficacy.', 'however these animal models have limitations due to species difference.', 'the viral tropism correlates to expression of viral receptor s on the surface of host cells and intracellular restriction as well.', 'cancer cell lines have been used widely for biological or functional cell systems in many other fields agarwal and rimm 2012 barretina et al 2012 palechor-ceron et al 2019 .', 'these cells are defective for many functions of normal human cells because of significant genetic and epigenetic changes compared to the normal tissues.', 'however primary cell yields and viability from any original tissue specimens are often relatively low or highly variable.', 'these normal cells cease at a few passages because of very limited proliferative capacity in vitro.', 'it has been extremely difficult for decades to generate and maintain normal cell lines.', 'several exogenous immortalization approaches have been used to establish primary cell lines from variety of tissue types.', 'for example viral oncogenes such as sv40 large t antigen or the human papillomavirus hpv e6/e7 proteins can bypass the cell senescence block through interfering with p53 and rb regulatory pathways liu et al 2005 liu et al 2007 liu et al 2008 cid arregui et al 2012 klingelhutz and roman 2012 ghittoni et al 2015 .', 'expression of exogenous cellular genes htert c-myc cdk4 etc or inactivation of cellular tumor repressors p53 prb etc can be widely used to immortalize primary human normal cells which leads to disrupted cell differentiation or loss of tissue type-specific functions liu et al 2012 .']\n",
            "['individuals with spinal cord injuries and disorders sci/d have an elevated risk of death with community-acquired pneumonia and there may be elevated risk with coronavirus disease 2019 covid-19 .', 'the veterans health administration vha sci/d national program office oversees health care services received by over 24 000 veterans with sci/d at 170 medical centers and more than 1000 outpatient care sites.', 'in coordination with the vha support service center the national program office maintains the vha sci/d registry which identifies veterans who have received health care services from the vha sci/d system of care.', 'to support outreach and prevention activities in the response to the current covid-19 pandemic the national program office developed operational reports showing covid-19 infection status as determined by occurrence of a positive laboratory test performed by vha or by entry of a non-vha lab result.', 'additional registry data including demographics neurologic classifications comorbidities and geographic setting were used to determine a preliminary infection risk category and severity risk category for each sci/d veteran.', 'these reports were developed in march 2020 and made available to vha sci/d centers on april 7 2020. the objective of this analysis was to describe case fatality of covid-19 infection in veterans with sci/d as determined using these operational reports.']\n",
            "['colorectal cancer crc is the fourth most common tumor and the fourth most common cause of tumor-associated mortality worldwide accounting for an estimated 1 4 million new cases and 700 000 deaths in 2012. the continued development of modalities for the early diagnosis and prevention of crc has tremendously benefited patient care.', 'however there were an estimated 1 8 million new cases of crc and 881 000 deaths in 2018 which accounted for about 10% of all newly diagnosed cancers and related deaths.', 'this tendency indicates the urgency for the early diagnosis and treatment of crc.', 'in addition the 5-year relative survival rate of crc patients remains at less than 50% in underdeveloped countries whereas the survival rate is approximately 65% in developed countries including canada the united states australia and several european countries 4–6 surgical resection plays a significant role in the management of colon cancer.', 'for stage iii colon adenocarcinoma coad adjuvant systemic treatment is crucial to decrease the incidence of recurrence and increase the overall survival os rate.', 'for patients with node-negative metastatic colon cancer administration of adjuvant systemic treatment can greatly decrease tumor recurrence and increase patient survival.', 'however the initiation of adjuvant systemic treatment is unnecessarily delayed in some cases.']\n",
            "['human rhinoviruses hrvs are highly prevalent respiratory pathogens traditionally associated with mild and self-limited upper respiratory tract infections ‘colds’ .', 'the replacement of relatively insensitive virus isolation methods with molecular methods for hrv detection has contributed to a recent reappraisal of their incidence of infection and genetic diversity.', 'hrvs are now known to be intimately linked to the development and exacerbations of respiratory diseases such as asthma khetsuriani et al 2008 miller et al 2007 and responsible for more severe disease manifestations such as bronchiolitis in young children and in the immunosuppressed gutman et al 2007 ison et al 2003 mcerlean et al 2007 .', 'genetic characterization of hrvs detected by molecular methods has revealed much greater diversity than previously described approximately one-third of hrv infections are now known to be caused by a third group of rhinoviruses species c uncultureable in vitro and therefore entirely missed by traditional virus isolation methods arden et al 2006 kaiser et al 2006 kistler et al 2007a lamson et al 2006 lau et al 2007 lee et al 2007 mcerlean et al 2007 olenec et al 2010 renwick et al 2007 .']\n",
            "['all viruses require host translational machinery in order to replicate.', 'to make use of this machinery while avoiding detection viruses have evolved a number of unconventional translational strategies gale et al 2000 including internal ribosome entry sites jang et al 1988 leaky scanning schwartz et al 1990 ribosome shunting schmidt-puchta et al 1997 reinitiation hemmings-mieszczak et al 1997 and programmed ribosomal frameshifting prf gesteland and atkins 1996 .', 'many of these strategies are unique to viruses and thus are potential antiviral targets harford 1995 .']\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "5"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 130
        }
      ]
    }
  ]
}